{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from datetime import date\n",
    "import glob\n",
    "import pandas as pd\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "today = date.today()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "files = f'data/02-out/*{today}.csv'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Read data\n",
    "\n",
    "Combining github and cran data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2\n",
      "625\n"
     ]
    }
   ],
   "source": [
    "df_arr = []\n",
    "for file in glob.glob(files):\n",
    "    df_arr.append(pd.read_csv(file))\n",
    "\n",
    "print(len(df_arr))\n",
    "df = pd.concat(df_arr)\n",
    "print(len(df))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['name', 'description', 'url', 'keywords', 'prog_lang', 'source',\n",
       "       'stars'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Remove duplicated packages cross-listed in python/R \n",
    "\n",
    "Treatment: if cross-listed, only count towards R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "dup = set(df[df.name.duplicated()]['name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_dups = pd.DataFrame()\n",
    "\n",
    "for w in dup:\n",
    "    x = df[df.name == w]\n",
    "    df_dups = df_dups.append(x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_dups.sort_values(['name', 'source']).to_csv(f'data/02-out/clean/pkgs-comb-{today}-dups.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.assign(duplicate='No')\n",
    "df.loc[(df.name.isin(dup)) & (df.source==\"Github\"), 'duplicate'] = 'Yes'\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check SDM and remove irrelevant packages\n",
    "Treatment: remove"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "200    Python implementation of nonparametric nearest-neighbor-based estimators for divergences between distributions.\n",
       "Name: description, dtype: object"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.set_option('display.max_colwidth', -1)\n",
    "pd.set_option('display.max_rows', 1000)\n",
    "df[df.name=='py-sdm']['description']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>192</th>\n",
       "      <td>SDMUSIC</td>\n",
       "      <td>Search &amp;&amp; Download Music from multi-platform</td>\n",
       "      <td>https://github.com/pcdack/SDMUSIC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193</th>\n",
       "      <td>menpofit</td>\n",
       "      <td>Menpo's 2D deformable modelling toolkit (AAMs/CLMs/SDMs)</td>\n",
       "      <td>https://github.com/menpo/menpofit</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194</th>\n",
       "      <td>landmark_py</td>\n",
       "      <td>Landmark with Regressition in Python （LBF(3000fps), ESR and SDM）</td>\n",
       "      <td>https://github.com/FacialLandmark/landmark_py</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>pandaSDMX</td>\n",
       "      <td>Python interface to SDMX</td>\n",
       "      <td>https://github.com/dr-leo/pandaSDMX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>Linux-SDM-Downloader</td>\n",
       "      <td>Downloader SDM files via SDX file (DreamSpark) on Linux</td>\n",
       "      <td>https://github.com/RadekSimkanic/Linux-SDM-Downloader</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>AANE_Python</td>\n",
       "      <td>Accelerated Attributed Network Embedding, SDM 2017</td>\n",
       "      <td>https://github.com/xhuang31/AANE_Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>py-sdm</td>\n",
       "      <td>Python implementation of nonparametric nearest-neighbor-based estimators for divergences between distributions.</td>\n",
       "      <td>https://github.com/dougalsutherland/py-sdm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>ArcSDM</td>\n",
       "      <td>Spatial Data Modeler 5 for ArcGis pro</td>\n",
       "      <td>https://github.com/gtkfi/ArcSDM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>SDML-Final</td>\n",
       "      <td>Transportation Mode Detection For HTC</td>\n",
       "      <td>https://github.com/BaiiYuan/SDML-Final</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>204</th>\n",
       "      <td>sdmpy-old</td>\n",
       "      <td>Updated May 19, 2016</td>\n",
       "      <td>https://github.com/caseyjlaw/sdmpy-old</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>sdml_final</td>\n",
       "      <td>Updated Jan 9, 2019</td>\n",
       "      <td>https://github.com/wheatdog/sdml_final</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>sdmp-python</td>\n",
       "      <td>Playing with implementing SDMP</td>\n",
       "      <td>https://github.com/jmhobbs/sdmp-python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>gen_sdm</td>\n",
       "      <td>Docs for the GEN project.</td>\n",
       "      <td>https://github.com/vinay87/gen_sdm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>sparkSDM</td>\n",
       "      <td>Updated Apr 23, 2018</td>\n",
       "      <td>https://github.com/tawjaw/sparkSDM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>Django-sdModels</td>\n",
       "      <td>Updated Jun 11, 2017</td>\n",
       "      <td>https://github.com/YunsikPark/Django-sdModels</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>iamc-sdmx-demo</td>\n",
       "      <td>IAMC data template implemented in SDMX</td>\n",
       "      <td>https://github.com/khaeru/iamc-sdmx-demo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>211</th>\n",
       "      <td>Deep-Co-Clustering</td>\n",
       "      <td>Code and data for Deep Co-Clustering (SDM'19)</td>\n",
       "      <td>https://github.com/DerronXu/Deep-Co-Clustering</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>sdm-lab-1</td>\n",
       "      <td>Updated Mar 13, 2019</td>\n",
       "      <td>https://github.com/haydarai/sdm-lab-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>SDM-Lab1</td>\n",
       "      <td>Updated Mar 13, 2019</td>\n",
       "      <td>https://github.com/PinarTurkyilmaz/SDM-Lab1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>214</th>\n",
       "      <td>SDMA-Community-Detection</td>\n",
       "      <td>Updated Oct 30, 2015</td>\n",
       "      <td>https://github.com/jaeindia/SDMA-Community-Detection</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>215</th>\n",
       "      <td>pipySDM-CD16</td>\n",
       "      <td>Python Raspberry pi SDMCD16 control</td>\n",
       "      <td>https://github.com/georgeredinger/pipySDM-CD16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216</th>\n",
       "      <td>srkim-py-sdmx</td>\n",
       "      <td>sdmx in python</td>\n",
       "      <td>https://github.com/seraekim/srkim-py-sdmx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217</th>\n",
       "      <td>sdmp-signed-yaml-python</td>\n",
       "      <td>An early Python implementation of sdmp/signed-yaml</td>\n",
       "      <td>https://github.com/jmhobbs/sdmp-signed-yaml-python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>SDM-PresencePreProc</td>\n",
       "      <td>Updated Feb 28, 2018</td>\n",
       "      <td>https://github.com/VANatHeritage/SDM-PresencePreProc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>project-euler-sdmx</td>\n",
       "      <td>Problems and Solutions for Project Euler</td>\n",
       "      <td>https://github.com/sdmx/project-euler-sdmx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220</th>\n",
       "      <td>sdm-word-segmentation</td>\n",
       "      <td>Using Sparse Distributed Memory for word segmentation. See if word boundaries can be discovered in highly-dimensional…</td>\n",
       "      <td>https://github.com/yczeng/sdm-word-segmentation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>cf_netSDM</td>\n",
       "      <td>Network analysis for semantic data mining</td>\n",
       "      <td>https://github.com/JanKralj/cf_netSDM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>sdm-lab-3</td>\n",
       "      <td>Updated May 6, 2019</td>\n",
       "      <td>https://github.com/haydarai/sdm-lab-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>SDM-ArcToolsEV</td>\n",
       "      <td>Toolbox and scripts used to produce environmental variables for SDM regional project</td>\n",
       "      <td>https://github.com/VANatHeritage/SDM-ArcToolsEV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>sdmr-paper-code</td>\n",
       "      <td>Updated Nov 22, 2016</td>\n",
       "      <td>https://github.com/charlienewey/sdmr-paper-code</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>SDM_Lab1</td>\n",
       "      <td>Updated Mar 13, 2019</td>\n",
       "      <td>https://github.com/hugrave-upc/SDM_Lab1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>ROB534_SDM</td>\n",
       "      <td>Updated Feb 14, 2019</td>\n",
       "      <td>https://github.com/YiHerngOng/ROB534_SDM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>228</th>\n",
       "      <td>Deep-Learning-SDM</td>\n",
       "      <td>As part of Sequential Decision Making course, this class project aims to leverage deep learning to solve the Sub-T ch…</td>\n",
       "      <td>https://github.com/abhiagni11/Deep-Learning-SDM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>229</th>\n",
       "      <td>openmm_sdm_workflow</td>\n",
       "      <td>Example of the use of the single-decoupling method setup workflow</td>\n",
       "      <td>https://github.com/egallicc/openmm_sdm_workflow</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230</th>\n",
       "      <td>project-SDM-Soil-Data-Monitoring-</td>\n",
       "      <td>this my project for PIMNAS 2018</td>\n",
       "      <td>https://github.com/imamhakim/project-SDM-Soil-Data-Monitoring-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>231</th>\n",
       "      <td>cmd-sdmx-to-v4</td>\n",
       "      <td>Tool to convert SDMX outputs to V4 format</td>\n",
       "      <td>https://github.com/ONS-OpenData/cmd-sdmx-to-v4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232</th>\n",
       "      <td>sdm-python-raspiSerialLib</td>\n",
       "      <td>Updated Sep 29, 2017</td>\n",
       "      <td>https://github.com/jm921106/sdm-python-raspiSerialLib</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233</th>\n",
       "      <td>upc-sdm-property-graphs-lab1</td>\n",
       "      <td>First mini project for Semantic Data Management on Property graphs using Neo4j, UPC, BDMA</td>\n",
       "      <td>https://github.com/ankushs92/upc-sdm-property-graphs-lab1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>CTMtoEMU</td>\n",
       "      <td>A python program to convert CTM files (usually generated by Kaldi) into an EMU SDMS database.</td>\n",
       "      <td>https://github.com/danijel3/CTMtoEMU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>truefis</td>\n",
       "      <td>Code and paper for an article presented at SDM'14 (and extended journal version in preparation)</td>\n",
       "      <td>https://github.com/rionda/truefis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>rfisweep</td>\n",
       "      <td>script collection using sdmpy to visualize peak amplitudes and switched power for VLA SDM datasets</td>\n",
       "      <td>https://github.com/fschinzel/rfisweep</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>AcceleratedExperimentalDesign</td>\n",
       "      <td>Code of paper \"Accelerated Experimental Design for Pairwise Comparisons\", by Guo et al., SDM 2019.</td>\n",
       "      <td>https://github.com/neu-spiral/AcceleratedExperimentalDesign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>fhv_paper_code</td>\n",
       "      <td>Code for the paper \"An LSTM approach to Patent Classification based on Fixed Hierarchy Vectors\" to be presented in SDM …</td>\n",
       "      <td>https://github.com/marawanokasha/fhv_paper_code</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239</th>\n",
       "      <td>spaudiopy</td>\n",
       "      <td>Spatial Audio Python Package</td>\n",
       "      <td>https://github.com/chris-hld/spaudiopy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>-</td>\n",
       "      <td>Binary Features）方法以及SDM（Supervised Decent Method）方法。但是，上述方法也存在一定的问题，例如，基于提取到的局部图像特征来定位时很难找到具有全局信息的关键点。基于点分布模型的方法和基于形…</td>\n",
       "      <td>https://github.com/1365843167/-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241</th>\n",
       "      <td>SDM3Dpose</td>\n",
       "      <td>Use Supervised Descent Method for 3D pose estimation.</td>\n",
       "      <td>https://github.com/ChrisYang/SDM3Dpose</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242</th>\n",
       "      <td>SDM630-Modbus</td>\n",
       "      <td>SDM630-Modbus 3P4W Multifunction Power Analyser WithRS485 Port modbus RTU for the domoticz plugin</td>\n",
       "      <td>https://github.com/MFxMF/SDM630-Modbus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>243</th>\n",
       "      <td>sdml-bandit-example</td>\n",
       "      <td>example implementation of a epsilon greedy bandit algorithm for algorithm workshop</td>\n",
       "      <td>https://github.com/brentpayne/sdml-bandit-example</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>244</th>\n",
       "      <td>sdmx.py</td>\n",
       "      <td>Read SDMX XML files using Python</td>\n",
       "      <td>https://github.com/mwilliamson/sdmx.py</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>SentimentAnalysisSDM</td>\n",
       "      <td>Implementation of sentiment analysis algorithms using labMT word set - and extending the set using Bayes classifier.</td>\n",
       "      <td>https://github.com/BogdanSorlea/SentimentAnalysisSDM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>sdmpy</td>\n",
       "      <td>Python classes for reading VLA/ALMA Science Data Model (SDM)</td>\n",
       "      <td>https://github.com/demorest/sdmpy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>247</th>\n",
       "      <td>widukind.py.pandaSDMX</td>\n",
       "      <td>Widukind connector for PandaSDMX</td>\n",
       "      <td>https://github.com/Widukind/widukind.py.pandaSDMX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248</th>\n",
       "      <td>sdm220-carbon</td>\n",
       "      <td>Simple script to get values from Eastron energy meter to Carbon</td>\n",
       "      <td>https://github.com/ranzhe/sdm220-carbon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249</th>\n",
       "      <td>LSCP</td>\n",
       "      <td>Supplementary material for SDM paper \"LSCP: Locally Selective Combination in Parallel Outlier Ensembles\"</td>\n",
       "      <td>https://github.com/yzhao062/LSCP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>250</th>\n",
       "      <td>sdmreader</td>\n",
       "      <td>Python toolkit for reading the radio astronomy data format known as the Science Data Model (SDM)</td>\n",
       "      <td>https://github.com/caseyjlaw/sdmreader</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>251</th>\n",
       "      <td>SDMS</td>\n",
       "      <td>Updated Dec 14, 2017</td>\n",
       "      <td>https://github.com/victor-gonzalez-navarro/SDMS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>252</th>\n",
       "      <td>SDMCEII</td>\n",
       "      <td>Specificity Determining Matrices of Co-Evolving Interaction Interfaces</td>\n",
       "      <td>https://github.com/Bowloforanges/SDMCEII</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>253</th>\n",
       "      <td>sdmc</td>\n",
       "      <td>software defined efficient multicast</td>\n",
       "      <td>https://github.com/prithviraj6116/sdmc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>254</th>\n",
       "      <td>SDM_Tools</td>\n",
       "      <td>Development environment for tools related to SDM prep and processing.</td>\n",
       "      <td>https://github.com/PNHP/SDM_Tools</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256</th>\n",
       "      <td>sdm2influx</td>\n",
       "      <td>read Eastron SDM 120/230 measurements via ModBus and store them to InfluxDB</td>\n",
       "      <td>https://github.com/lesinigo/sdm2influx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257</th>\n",
       "      <td>sdml</td>\n",
       "      <td>Smokey feedback-based realtime machine learning thing.</td>\n",
       "      <td>https://github.com/Charcoal-SE/sdml</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>258</th>\n",
       "      <td>TIB-RDFizer</td>\n",
       "      <td>SDM RDFizer</td>\n",
       "      <td>https://github.com/SDM-TIB/TIB-RDFizer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259</th>\n",
       "      <td>sdmplayer-kt</td>\n",
       "      <td>SotongDJ mplayer control script set for Kindle Touch Launcher</td>\n",
       "      <td>https://github.com/SotongDJ/sdmplayer-kt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260</th>\n",
       "      <td>xSDM-python-wrapper</td>\n",
       "      <td>Python wrapper for xSDM utility</td>\n",
       "      <td>https://github.com/exander77/xSDM-python-wrapper</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261</th>\n",
       "      <td>sdm630</td>\n",
       "      <td>SDM630 - modbus</td>\n",
       "      <td>https://github.com/coolex/sdm630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>262</th>\n",
       "      <td>SDMiniTools</td>\n",
       "      <td>Collection of small scripts &amp; tools</td>\n",
       "      <td>https://github.com/SDBagel/SDMiniTools</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>263</th>\n",
       "      <td>SDM630</td>\n",
       "      <td>Python library for reading data from SDM630 with MQTT interface</td>\n",
       "      <td>https://github.com/xtompok/SDM630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>264</th>\n",
       "      <td>sdm2</td>\n",
       "      <td>Updated Apr 29, 2017</td>\n",
       "      <td>https://github.com/binderchri/sdm2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>265</th>\n",
       "      <td>hedwig3</td>\n",
       "      <td>Towards scalable SDM (Hedwig update)</td>\n",
       "      <td>https://github.com/SkBlaz/hedwig3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>266</th>\n",
       "      <td>SDMA2</td>\n",
       "      <td>MIT license</td>\n",
       "      <td>https://github.com/develope126/SDMA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267</th>\n",
       "      <td>home-assistant_sdm630</td>\n",
       "      <td>EASTON SDM630</td>\n",
       "      <td>https://github.com/ticapix/home-assistant_sdm630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268</th>\n",
       "      <td>DBMSTClu</td>\n",
       "      <td>Code related to SDM 2018 Paper \"Graph sketching-based Space-efficient Data Clustering\"</td>\n",
       "      <td>https://github.com/annemorvan/DBMSTClu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>269</th>\n",
       "      <td>sdm630-report</td>\n",
       "      <td>Tools for monitoring an SDM630 3 phase meter and reporting output to PVoutput</td>\n",
       "      <td>https://github.com/ruddj/sdm630-report</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270</th>\n",
       "      <td>fkl-SDM16</td>\n",
       "      <td>Code and experiments for \"A confidence-based approach for balancing fairness and accuracy\"</td>\n",
       "      <td>https://github.com/j2kun/fkl-SDM16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271</th>\n",
       "      <td>SDMinesLabDisplay</td>\n",
       "      <td>A variety of different display based OpenCV programs for the SD Mines UAS and Robotics Labs.</td>\n",
       "      <td>https://github.com/PhilTKamp/SDMinesLabDisplay</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>272</th>\n",
       "      <td>sdm220_eastron_wattmeter</td>\n",
       "      <td>GPL-3.0 license</td>\n",
       "      <td>https://github.com/nickma82/sdm220_eastron_wattmeter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273</th>\n",
       "      <td>SDMAdressBookCreator</td>\n",
       "      <td>A python script that reads a google form results and parses the results into a CSV file. The data compatible with Dev…</td>\n",
       "      <td>https://github.com/pemby/SDMAdressBookCreator</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>274</th>\n",
       "      <td>py-SDM</td>\n",
       "      <td>General python scripts for SDM projects</td>\n",
       "      <td>https://github.com/VANatHeritage/py-SDM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275</th>\n",
       "      <td>SDML</td>\n",
       "      <td>NTU CSIE 2018 Fall, Special Topics/Directions on Machine Leaning (SDML)</td>\n",
       "      <td>https://github.com/victai/SDML</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>276</th>\n",
       "      <td>SDMXPlugin</td>\n",
       "      <td>QGIS Plugin to query SDMX data cubes served as WFS feature types</td>\n",
       "      <td>https://github.com/lmorandini/SDMXPlugin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>277</th>\n",
       "      <td>sdmb</td>\n",
       "      <td>Basic dream-blog</td>\n",
       "      <td>https://github.com/Raveline/sdmb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>278</th>\n",
       "      <td>SDMini</td>\n",
       "      <td>Senior Design Mini Project</td>\n",
       "      <td>https://github.com/thakkarV/SDMini</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>sdmgr</td>\n",
       "      <td>Simple Daemon Manager</td>\n",
       "      <td>https://github.com/avl93/sdmgr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>282</th>\n",
       "      <td>SDMA</td>\n",
       "      <td>Updated Dec 18, 2016</td>\n",
       "      <td>https://github.com/hazemsoliman115/SDMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>283</th>\n",
       "      <td>SDMPPO</td>\n",
       "      <td>Encryption algorithm based on circle geometry.</td>\n",
       "      <td>https://github.com/Frajerzycki/SDMPPO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>Python-Treadmill-Prototype</td>\n",
       "      <td>Python code prototyping ANT+ SDM Sensor</td>\n",
       "      <td>https://github.com/jonathanpandres/Python-Treadmill-Prototype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285</th>\n",
       "      <td>csv_sdmx</td>\n",
       "      <td>Updated Apr 1, 2019</td>\n",
       "      <td>https://github.com/workingravi/csv_sdmx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>sdg-sdmx</td>\n",
       "      <td>Converting sdg data to SDMX</td>\n",
       "      <td>https://github.com/ONSdigital/sdg-sdmx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287</th>\n",
       "      <td>sdm_project</td>\n",
       "      <td>Final Project: ROB534 - Sequential Decision Making in Robotics</td>\n",
       "      <td>https://github.com/dietriro/sdm_project</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>288</th>\n",
       "      <td>sdm-docker</td>\n",
       "      <td>Docker build files for scientific data management</td>\n",
       "      <td>https://github.com/suchyta1/sdm-docker</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>289</th>\n",
       "      <td>SDM_CPAB</td>\n",
       "      <td>Updated Nov 17, 2016</td>\n",
       "      <td>https://github.com/miguelgamendes/SDM_CPAB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290</th>\n",
       "      <td>sdm-project</td>\n",
       "      <td>It's a course final project which is developed in a course name 'Software Develop Method' in NTUIM.</td>\n",
       "      <td>https://github.com/weichih-c/sdm-project</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>291</th>\n",
       "      <td>sdmpv_bot</td>\n",
       "      <td>GPL-3.0 license</td>\n",
       "      <td>https://github.com/dbaldanza/sdmpv_bot</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>292</th>\n",
       "      <td>PICMath_SDMA</td>\n",
       "      <td>Updated Apr 23, 2019</td>\n",
       "      <td>https://github.com/TNWOG/PICMath_SDMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>bayes-sdm</td>\n",
       "      <td>Updated Dec 15, 2012</td>\n",
       "      <td>https://github.com/berkeley-cocosci/bayes-sdm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>sdmx-converter</td>\n",
       "      <td>Python scripts to convert SDMX files to CSV. Initially only for Statscan Census files.</td>\n",
       "      <td>https://github.com/plot-and-scatter/sdmx-converter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295</th>\n",
       "      <td>WASHWoSDMDA</td>\n",
       "      <td>WASH WoS Data Management and Data Analysis</td>\n",
       "      <td>https://github.com/udaraz77/WASHWoSDMDA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>SDM_4</td>\n",
       "      <td>Updated Dec 12, 2017</td>\n",
       "      <td>https://github.com/mike840609/SDM_4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>298</th>\n",
       "      <td>sdmxBrowser</td>\n",
       "      <td>UI for RJSDMX</td>\n",
       "      <td>https://github.com/bowerth/sdmxBrowser</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>300</th>\n",
       "      <td>modules</td>\n",
       "      <td>modules for each component of an SDM workflow, for use with the zoon R package</td>\n",
       "      <td>https://github.com/zoonproject/modules</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301</th>\n",
       "      <td>hSDM_Tutorial</td>\n",
       "      <td>Updated Feb 27, 2015</td>\n",
       "      <td>https://github.com/adammwilson/hSDM_Tutorial</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>302</th>\n",
       "      <td>PointedSDMs</td>\n",
       "      <td>Updated Aug 16, 2017</td>\n",
       "      <td>https://github.com/oharar/PointedSDMs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304</th>\n",
       "      <td>QTP_SDM</td>\n",
       "      <td>Updated Mar 20, 2019</td>\n",
       "      <td>https://github.com/saltpawpaw/QTP_SDM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>305</th>\n",
       "      <td>SDM_class</td>\n",
       "      <td>Updated Mar 19, 2018</td>\n",
       "      <td>https://github.com/mmsmith9/SDM_class</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>Mongolia_SDM</td>\n",
       "      <td>Updated Jan 19, 2017</td>\n",
       "      <td>https://github.com/gallington/Mongolia_SDM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310</th>\n",
       "      <td>SDM-Project</td>\n",
       "      <td>Updated Dec 29, 2018</td>\n",
       "      <td>https://github.com/freyagenesis/SDM-Project</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311</th>\n",
       "      <td>BBS.SDM</td>\n",
       "      <td>Updated Oct 19, 2018</td>\n",
       "      <td>https://github.com/RushingLab/BBS.SDM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>312</th>\n",
       "      <td>Establishment_SDMs</td>\n",
       "      <td>Updated Jul 11, 2016</td>\n",
       "      <td>https://github.com/crollinson/Establishment_SDMs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314</th>\n",
       "      <td>sdm_analysis</td>\n",
       "      <td>Analysis of the Social Decision Making Algorithms</td>\n",
       "      <td>https://github.com/rpitrust/sdm_analysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316</th>\n",
       "      <td>Malaria_SDM</td>\n",
       "      <td>Updated Mar 19, 2019</td>\n",
       "      <td>https://github.com/KIT-IfGG/Malaria_SDM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>318</th>\n",
       "      <td>sdmengine.common</td>\n",
       "      <td>Updated Sep 4, 2015</td>\n",
       "      <td>https://github.com/howl-anderson/sdmengine.common</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>320</th>\n",
       "      <td>SDM_traits</td>\n",
       "      <td>Updated Apr 18, 2016</td>\n",
       "      <td>https://github.com/KristinaRiemer/SDM_traits</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>321</th>\n",
       "      <td>bbs-SDM</td>\n",
       "      <td>Updated Oct 20, 2017</td>\n",
       "      <td>https://github.com/sdtaylor/bbs-SDM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>322</th>\n",
       "      <td>SDM-comparison</td>\n",
       "      <td>Updated Apr 11, 2019</td>\n",
       "      <td>https://github.com/aminorberg/SDM-comparison</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>323</th>\n",
       "      <td>sdm_biofungicide</td>\n",
       "      <td>results from my biofungicide trials that I ran in the Salinas/Pajaro valley for spinach downy mildew in 2013-2016.</td>\n",
       "      <td>https://github.com/robchoudhury/sdm_biofungicide</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>324</th>\n",
       "      <td>SDMA_Exercises</td>\n",
       "      <td>Statistics, Data Mining and Apprentissage Exercises</td>\n",
       "      <td>https://github.com/KK-Projects/SDMA_Exercises</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>325</th>\n",
       "      <td>nhSDM</td>\n",
       "      <td>R package with useful functions for working with SDMs and Natural Heritage data</td>\n",
       "      <td>https://github.com/VANatHeritage/nhSDM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>327</th>\n",
       "      <td>sdmengine.dismo</td>\n",
       "      <td>Updated Sep 4, 2015</td>\n",
       "      <td>https://github.com/howl-anderson/sdmengine.dismo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>sdm_dalbergias</td>\n",
       "      <td>Proyecto de modelación de distribución potencial de Dalbergias</td>\n",
       "      <td>https://github.com/fpardourrutia/sdm_dalbergias</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>330</th>\n",
       "      <td>AMF_SDMs</td>\n",
       "      <td>Updated Aug 27, 2018</td>\n",
       "      <td>https://github.com/bobmuscarella/AMF_SDMs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>331</th>\n",
       "      <td>BV-SDM_automator</td>\n",
       "      <td>R script to automate generation of SDM files</td>\n",
       "      <td>https://github.com/Joe-Hilgard/BV-SDM_automator</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>332</th>\n",
       "      <td>MinBAR</td>\n",
       "      <td>Package to calculate Minimum Background Area for SDMs</td>\n",
       "      <td>https://github.com/xavi-rp/MinBAR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333</th>\n",
       "      <td>SDM_ENSO_Butterflies</td>\n",
       "      <td>code used in manuscript - Potential impacts of a future persistent El Nino or La Nina on three subspecies of Australi…</td>\n",
       "      <td>https://github.com/daisyduursma/SDM_ENSO_Butterflies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334</th>\n",
       "      <td>SDM_course_git</td>\n",
       "      <td>Updated May 1, 2019</td>\n",
       "      <td>https://github.com/KatieMCook/SDM_course_git</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>335</th>\n",
       "      <td>Severn-Estuary-SDMS</td>\n",
       "      <td>Project modelling the change of six species within the Severn Estuary and the impact of the Severn Barrage and Cardif…</td>\n",
       "      <td>https://github.com/laxmack21/Severn-Estuary-SDMS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>337</th>\n",
       "      <td>Local-Ecological-knowledge-Model-evaluation</td>\n",
       "      <td>Use local ecological knowledge to quantitatively evalaute SDM outputs</td>\n",
       "      <td>https://github.com/torrem/Local-Ecological-knowledge-Model-evaluation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338</th>\n",
       "      <td>beech-fir-sdm</td>\n",
       "      <td>Updated Mar 28, 2017</td>\n",
       "      <td>https://github.com/lukasbaumbach/beech-fir-sdm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340</th>\n",
       "      <td>biodiversity-sdm-lesson</td>\n",
       "      <td>Introductory lesson to generate range maps for butterfly-host plant interactions and predict distributional shifts us…</td>\n",
       "      <td>https://github.com/jcoliver/biodiversity-sdm-lesson</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>341</th>\n",
       "      <td>simple_sdm_pipeline</td>\n",
       "      <td>A simple example of how to use drake to create a pipeline for SDMs.</td>\n",
       "      <td>https://github.com/jequihua/simple_sdm_pipeline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343</th>\n",
       "      <td>2016_Ecology_SDM</td>\n",
       "      <td>201603_Ecology</td>\n",
       "      <td>https://github.com/kanji14134/2016_Ecology_SDM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>344</th>\n",
       "      <td>coral_traits_SDM</td>\n",
       "      <td>Updated Jul 20, 2015</td>\n",
       "      <td>https://github.com/tlmizerek/coral_traits_SDM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345</th>\n",
       "      <td>SDM.Virtual.Species_Bell.Schlaepfer</td>\n",
       "      <td>Code for generating virtual species and testing SDM performance</td>\n",
       "      <td>https://github.com/bellland/SDM.Virtual.Species_Bell.Schlaepfer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>347</th>\n",
       "      <td>SDM-batch-scripts</td>\n",
       "      <td>AGPL-3.0 license</td>\n",
       "      <td>https://github.com/jallen73/SDM-batch-scripts</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>349</th>\n",
       "      <td>SGH_SDM_18</td>\n",
       "      <td>Statistical Decision Methods at SGH, summer semester 2017/18</td>\n",
       "      <td>https://github.com/nosarzewski/SGH_SDM_18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>350</th>\n",
       "      <td>sdm.nlu2016ss</td>\n",
       "      <td>Updated Apr 21, 2016</td>\n",
       "      <td>https://github.com/dschlaep/sdm.nlu2016ss</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>352</th>\n",
       "      <td>speciesdistributionmodeling</td>\n",
       "      <td>General scripts necessary for steps during the SDM approach</td>\n",
       "      <td>https://github.com/Rilquer/speciesdistributionmodeling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353</th>\n",
       "      <td>DynamicEnsembleSDM</td>\n",
       "      <td>Model fitting and evaluation for dynamic ensemble blue whale distribution model</td>\n",
       "      <td>https://github.com/briana-abrahms/DynamicEnsembleSDM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>354</th>\n",
       "      <td>Plant_SDM_code</td>\n",
       "      <td>Updated May 10, 2019</td>\n",
       "      <td>https://github.com/HMB3/Plant_SDM_code</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>355</th>\n",
       "      <td>SDM-Model-FIFA-19</td>\n",
       "      <td>The Repository Models FIFA 19 Players and gives an insight on what traits in a Player is important in judging the Ove…</td>\n",
       "      <td>https://github.com/aashishkakar/SDM-Model-FIFA-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>356</th>\n",
       "      <td>Lyon.MS.Thesis-SDM.Project</td>\n",
       "      <td>A version of Nicholas J Lyon's modeling thesis chapter (SDMs) that is ready to be made publicly available post-public…</td>\n",
       "      <td>https://github.com/NJLyon-Projects/Lyon.MS.Thesis-SDM.Project</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>357</th>\n",
       "      <td>seegSDM-bias-testing-fixture</td>\n",
       "      <td>A basic test fixture for testing the impact of different absence data approaches in seegSDM</td>\n",
       "      <td>https://github.com/fshearer/seegSDM-bias-testing-fixture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>358</th>\n",
       "      <td>BioScen1.5_SDM</td>\n",
       "      <td>Updated Apr 25, 2019</td>\n",
       "      <td>https://github.com/christianhof/BioScen1.5_SDM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>360</th>\n",
       "      <td>bbs</td>\n",
       "      <td>Various bbs related modeling I'm doing. One of them, a meta analysis of bbs SDM's, is now in rep bbs-SDM</td>\n",
       "      <td>https://github.com/sdtaylor/bbs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361</th>\n",
       "      <td>STManaged</td>\n",
       "      <td>Spatially explicit version of a State and Transition Model with forest management</td>\n",
       "      <td>https://github.com/willvieira/STManaged</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>362</th>\n",
       "      <td>Predictor_decomposition</td>\n",
       "      <td>Working scripts for creating SDMs and decomposing predictors to assess variable importance under climate change</td>\n",
       "      <td>https://github.com/naiamh/Predictor_decomposition</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>363</th>\n",
       "      <td>oak_gap_analysis</td>\n",
       "      <td>Compilation of ex situ collections, and in situ occurrence points; creation of SDMs</td>\n",
       "      <td>https://github.com/esbeckman/oak_gap_analysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>364</th>\n",
       "      <td>checkyourself</td>\n",
       "      <td>Repository for the SDM simulation model associated with Yanco et al. 2018</td>\n",
       "      <td>https://github.com/syanco/checkyourself</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365</th>\n",
       "      <td>INFO522Project</td>\n",
       "      <td>INFO522 proejct repository cloned from https://github.com/jcoliver/ebutterfly-sdm</td>\n",
       "      <td>https://github.com/kensaku-okada/INFO522Project</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>366</th>\n",
       "      <td>Mapping_phylogenetic_diversity</td>\n",
       "      <td>Scripts to map phylogenetic diversity from IUCN maps or SDM´s</td>\n",
       "      <td>https://github.com/lgchumboldt/Mapping_phylogenetic_diversity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>367</th>\n",
       "      <td>PROs</td>\n",
       "      <td>Methods and code for assembling the State Lands PROs: polygon processing, SDMs, hypergrid, polygon/raster products</td>\n",
       "      <td>https://github.com/NYNatHeritage/PROs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>368</th>\n",
       "      <td>shiny_STM-managed</td>\n",
       "      <td>Shiny App for the analytical analysis of the State and Transition Model with forest management</td>\n",
       "      <td>https://github.com/willvieira/shiny_STM-managed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>369</th>\n",
       "      <td>GIST_Thesis</td>\n",
       "      <td>Thesis for GIST master’s at USC-SSCI. Study compares tuned versus default MaxEnt models at two different covariate re…</td>\n",
       "      <td>https://github.com/CassKal/GIST_Thesis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>371</th>\n",
       "      <td>ebutterfly-sdm</td>\n",
       "      <td>MIT license</td>\n",
       "      <td>https://github.com/jcoliver/ebutterfly-sdm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>372</th>\n",
       "      <td>sdmTMB</td>\n",
       "      <td>An R package for spatial and spatiotemporal GLMMs with TMB</td>\n",
       "      <td>https://github.com/pbs-assess/sdmTMB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>377</th>\n",
       "      <td>rmaxent</td>\n",
       "      <td>Tools for working with Maxent in R</td>\n",
       "      <td>https://github.com/johnbaums/rmaxent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>378</th>\n",
       "      <td>SDMpriors</td>\n",
       "      <td>Exploring the potential to incorporate physiological priors in species distribution models</td>\n",
       "      <td>https://github.com/HuckleyLab/SDMpriors</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>380</th>\n",
       "      <td>SDM-CLMcomp</td>\n",
       "      <td>Scripts used in Maguire et al....</td>\n",
       "      <td>https://github.com/fitzLab-AL/SDM-CLMcomp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>381</th>\n",
       "      <td>eSDM</td>\n",
       "      <td>A user-friendly spatial tool with a web-based interface that allows users to import SDM layers and create and explore…</td>\n",
       "      <td>https://github.com/smwoodman/eSDM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>383</th>\n",
       "      <td>NaiveBayes</td>\n",
       "      <td>SDM</td>\n",
       "      <td>https://github.com/swapnilawasthi/NaiveBayes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>384</th>\n",
       "      <td>SDM-teaching</td>\n",
       "      <td>Teach SDM</td>\n",
       "      <td>https://github.com/madanksuwal/SDM-teaching</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>385</th>\n",
       "      <td>SDMproject</td>\n",
       "      <td>SDM project</td>\n",
       "      <td>https://github.com/swapnilawasthi/SDMproject</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>386</th>\n",
       "      <td>Composition</td>\n",
       "      <td>SDMs spatial SDM composition using an automatic R Script</td>\n",
       "      <td>https://github.com/ccsosa/Composition</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>389</th>\n",
       "      <td>Statistical_Data_Mining-Final_Project</td>\n",
       "      <td>SDM final Project</td>\n",
       "      <td>https://github.com/gauravjetley/Statistical_Data_Mining-Final_Project</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>390</th>\n",
       "      <td>244_SMLW</td>\n",
       "      <td>Intro, SDM, Island Variables</td>\n",
       "      <td>https://github.com/alyssa-a-winchell/244_SMLW</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>391</th>\n",
       "      <td>PSM-Model</td>\n",
       "      <td>PSM Modeling for SDM</td>\n",
       "      <td>https://github.com/mcolvin/PSM-Model</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>392</th>\n",
       "      <td>Light-Pollution-Project</td>\n",
       "      <td>SDM Final Project</td>\n",
       "      <td>https://github.com/sonia3187/Light-Pollution-Project</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>393</th>\n",
       "      <td>SDM-Final-Project</td>\n",
       "      <td>NCKU SDM Final Project</td>\n",
       "      <td>https://github.com/maciao/SDM-Final-Project</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>394</th>\n",
       "      <td>SDMclass2018</td>\n",
       "      <td>Updated Mar 19, 2018</td>\n",
       "      <td>https://github.com/kthompson25/SDMclass2018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395</th>\n",
       "      <td>SDMexoticPlants</td>\n",
       "      <td>code used to create outputs for weedfutures.net and article: Next-generation invaders? Hotspots for naturalised sleep…</td>\n",
       "      <td>https://github.com/daisyduursma/SDMexoticPlants</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>Makrooekologie</td>\n",
       "      <td>Kurs Makroökologie, erstellen von SDM</td>\n",
       "      <td>https://github.com/debroize/Makrooekologie</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>397</th>\n",
       "      <td>sdmhw5</td>\n",
       "      <td>Naive Bayes, Recursive partition tree and gradient boosting model</td>\n",
       "      <td>https://github.com/swapnilawasthi/sdmhw5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>398</th>\n",
       "      <td>SDMProjectA</td>\n",
       "      <td>Updated Apr 8, 2019</td>\n",
       "      <td>https://github.com/hayeshursh/SDMProjectA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399</th>\n",
       "      <td>ECOP-VLab-SDM</td>\n",
       "      <td>ECOPOTENTIAL IRIS SDM model/service</td>\n",
       "      <td>https://github.com/joaofgoncalves/ECOP-VLab-SDM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>400</th>\n",
       "      <td>STA546_SDM2</td>\n",
       "      <td>Contains code for assignments in UB STA546 Course</td>\n",
       "      <td>https://github.com/varadtupe/STA546_SDM2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>401</th>\n",
       "      <td>SDMDataAnalyticsProject</td>\n",
       "      <td>Updated Jan 12, 2019</td>\n",
       "      <td>https://github.com/rahilshaik/SDMDataAnalyticsProject</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>402</th>\n",
       "      <td>landcover-sdms</td>\n",
       "      <td>Scripts for sdm post processing using landcover</td>\n",
       "      <td>https://github.com/LBAB-Humboldt/landcover-sdms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>404</th>\n",
       "      <td>things</td>\n",
       "      <td>Miscellaneous R functions - mainly spatial and SDM</td>\n",
       "      <td>https://github.com/johnbaums/things</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>405</th>\n",
       "      <td>SDMPlay</td>\n",
       "      <td>Updated Oct 10, 2017</td>\n",
       "      <td>https://github.com/charleneguillaumot/SDMPlay</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406</th>\n",
       "      <td>SDMMetrics</td>\n",
       "      <td>Updated Jun 7, 2018</td>\n",
       "      <td>https://github.com/Farewe/SDMMetrics</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>407</th>\n",
       "      <td>sdmstack</td>\n",
       "      <td>Updated Jul 20, 2016</td>\n",
       "      <td>https://github.com/davharris/sdmstack</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>408</th>\n",
       "      <td>SDMbase</td>\n",
       "      <td>GPL-3.0 license</td>\n",
       "      <td>https://github.com/RWunderlich/SDMbase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>SDMXWrappers</td>\n",
       "      <td>A collection of wrappers on top of RJSDMX package</td>\n",
       "      <td>https://github.com/jcizel/SDMXWrappers</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>410</th>\n",
       "      <td>SDMstarter</td>\n",
       "      <td>Updated Apr 8, 2019</td>\n",
       "      <td>https://github.com/Lin-Jing-R/SDMstarter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>411</th>\n",
       "      <td>SapienDistributionModel</td>\n",
       "      <td>Debiasing of SDM using Social Media Data</td>\n",
       "      <td>https://github.com/guyawn/SapienDistributionModel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>413</th>\n",
       "      <td>sdmhr</td>\n",
       "      <td>Updated Sep 12, 2014</td>\n",
       "      <td>https://github.com/grgurev/sdmhr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>415</th>\n",
       "      <td>sdmflow</td>\n",
       "      <td>Updated Feb 8, 2019</td>\n",
       "      <td>https://github.com/juoe/sdmflow</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419</th>\n",
       "      <td>sdmanalysis</td>\n",
       "      <td>Updated Sep 4, 2015</td>\n",
       "      <td>https://github.com/howl-anderson/sdmanalysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>sdmid</td>\n",
       "      <td>Repository for blog code</td>\n",
       "      <td>https://github.com/daniel-rodak/sdmid</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>422</th>\n",
       "      <td>RobiniaSDM</td>\n",
       "      <td>Species distribution modeling of the nitrogen-fixing tree black locust</td>\n",
       "      <td>https://github.com/JeffreyMinucci/RobiniaSDM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424</th>\n",
       "      <td>tapir_sdms</td>\n",
       "      <td>Dissertation Research</td>\n",
       "      <td>https://github.com/codyschank/tapir_sdms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>BioticSDMs</td>\n",
       "      <td>Updated Mar 3, 2016</td>\n",
       "      <td>https://github.com/katieleach/BioticSDMs</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            name  \\\n",
       "192  SDMUSIC                                       \n",
       "193  menpofit                                      \n",
       "194  landmark_py                                   \n",
       "195  pandaSDMX                                     \n",
       "197  Linux-SDM-Downloader                          \n",
       "199  AANE_Python                                   \n",
       "200  py-sdm                                        \n",
       "201  ArcSDM                                        \n",
       "202  SDML-Final                                    \n",
       "204  sdmpy-old                                     \n",
       "205  sdml_final                                    \n",
       "206  sdmp-python                                   \n",
       "207  gen_sdm                                       \n",
       "208  sparkSDM                                      \n",
       "209  Django-sdModels                               \n",
       "210  iamc-sdmx-demo                                \n",
       "211  Deep-Co-Clustering                            \n",
       "212  sdm-lab-1                                     \n",
       "213  SDM-Lab1                                      \n",
       "214  SDMA-Community-Detection                      \n",
       "215  pipySDM-CD16                                  \n",
       "216  srkim-py-sdmx                                 \n",
       "217  sdmp-signed-yaml-python                       \n",
       "218  SDM-PresencePreProc                           \n",
       "219  project-euler-sdmx                            \n",
       "220  sdm-word-segmentation                         \n",
       "221  cf_netSDM                                     \n",
       "222  sdm-lab-3                                     \n",
       "223  SDM-ArcToolsEV                                \n",
       "224  sdmr-paper-code                               \n",
       "225  SDM_Lab1                                      \n",
       "226  ROB534_SDM                                    \n",
       "228  Deep-Learning-SDM                             \n",
       "229  openmm_sdm_workflow                           \n",
       "230  project-SDM-Soil-Data-Monitoring-             \n",
       "231  cmd-sdmx-to-v4                                \n",
       "232  sdm-python-raspiSerialLib                     \n",
       "233  upc-sdm-property-graphs-lab1                  \n",
       "234  CTMtoEMU                                      \n",
       "235  truefis                                       \n",
       "236  rfisweep                                      \n",
       "237  AcceleratedExperimentalDesign                 \n",
       "238  fhv_paper_code                                \n",
       "239  spaudiopy                                     \n",
       "240  -                                             \n",
       "241  SDM3Dpose                                     \n",
       "242  SDM630-Modbus                                 \n",
       "243  sdml-bandit-example                           \n",
       "244  sdmx.py                                       \n",
       "245  SentimentAnalysisSDM                          \n",
       "246  sdmpy                                         \n",
       "247  widukind.py.pandaSDMX                         \n",
       "248  sdm220-carbon                                 \n",
       "249  LSCP                                          \n",
       "250  sdmreader                                     \n",
       "251  SDMS                                          \n",
       "252  SDMCEII                                       \n",
       "253  sdmc                                          \n",
       "254  SDM_Tools                                     \n",
       "256  sdm2influx                                    \n",
       "257  sdml                                          \n",
       "258  TIB-RDFizer                                   \n",
       "259  sdmplayer-kt                                  \n",
       "260  xSDM-python-wrapper                           \n",
       "261  sdm630                                        \n",
       "262  SDMiniTools                                   \n",
       "263  SDM630                                        \n",
       "264  sdm2                                          \n",
       "265  hedwig3                                       \n",
       "266  SDMA2                                         \n",
       "267  home-assistant_sdm630                         \n",
       "268  DBMSTClu                                      \n",
       "269  sdm630-report                                 \n",
       "270  fkl-SDM16                                     \n",
       "271  SDMinesLabDisplay                             \n",
       "272  sdm220_eastron_wattmeter                      \n",
       "273  SDMAdressBookCreator                          \n",
       "274  py-SDM                                        \n",
       "275  SDML                                          \n",
       "276  SDMXPlugin                                    \n",
       "277  sdmb                                          \n",
       "278  SDMini                                        \n",
       "281  sdmgr                                         \n",
       "282  SDMA                                          \n",
       "283  SDMPPO                                        \n",
       "284  Python-Treadmill-Prototype                    \n",
       "285  csv_sdmx                                      \n",
       "286  sdg-sdmx                                      \n",
       "287  sdm_project                                   \n",
       "288  sdm-docker                                    \n",
       "289  SDM_CPAB                                      \n",
       "290  sdm-project                                   \n",
       "291  sdmpv_bot                                     \n",
       "292  PICMath_SDMA                                  \n",
       "293  bayes-sdm                                     \n",
       "294  sdmx-converter                                \n",
       "295  WASHWoSDMDA                                   \n",
       "296  SDM_4                                         \n",
       "298  sdmxBrowser                                   \n",
       "300  modules                                       \n",
       "301  hSDM_Tutorial                                 \n",
       "302  PointedSDMs                                   \n",
       "304  QTP_SDM                                       \n",
       "305  SDM_class                                     \n",
       "307  Mongolia_SDM                                  \n",
       "310  SDM-Project                                   \n",
       "311  BBS.SDM                                       \n",
       "312  Establishment_SDMs                            \n",
       "314  sdm_analysis                                  \n",
       "316  Malaria_SDM                                   \n",
       "318  sdmengine.common                              \n",
       "320  SDM_traits                                    \n",
       "321  bbs-SDM                                       \n",
       "322  SDM-comparison                                \n",
       "323  sdm_biofungicide                              \n",
       "324  SDMA_Exercises                                \n",
       "325  nhSDM                                         \n",
       "327  sdmengine.dismo                               \n",
       "328  sdm_dalbergias                                \n",
       "330  AMF_SDMs                                      \n",
       "331  BV-SDM_automator                              \n",
       "332  MinBAR                                        \n",
       "333  SDM_ENSO_Butterflies                          \n",
       "334  SDM_course_git                                \n",
       "335  Severn-Estuary-SDMS                           \n",
       "337  Local-Ecological-knowledge-Model-evaluation   \n",
       "338  beech-fir-sdm                                 \n",
       "340  biodiversity-sdm-lesson                       \n",
       "341  simple_sdm_pipeline                           \n",
       "343  2016_Ecology_SDM                              \n",
       "344  coral_traits_SDM                              \n",
       "345  SDM.Virtual.Species_Bell.Schlaepfer           \n",
       "347  SDM-batch-scripts                             \n",
       "349  SGH_SDM_18                                    \n",
       "350  sdm.nlu2016ss                                 \n",
       "352  speciesdistributionmodeling                   \n",
       "353  DynamicEnsembleSDM                            \n",
       "354  Plant_SDM_code                                \n",
       "355  SDM-Model-FIFA-19                             \n",
       "356  Lyon.MS.Thesis-SDM.Project                    \n",
       "357  seegSDM-bias-testing-fixture                  \n",
       "358  BioScen1.5_SDM                                \n",
       "360  bbs                                           \n",
       "361  STManaged                                     \n",
       "362  Predictor_decomposition                       \n",
       "363  oak_gap_analysis                              \n",
       "364  checkyourself                                 \n",
       "365  INFO522Project                                \n",
       "366  Mapping_phylogenetic_diversity                \n",
       "367  PROs                                          \n",
       "368  shiny_STM-managed                             \n",
       "369  GIST_Thesis                                   \n",
       "371  ebutterfly-sdm                                \n",
       "372  sdmTMB                                        \n",
       "377  rmaxent                                       \n",
       "378  SDMpriors                                     \n",
       "380  SDM-CLMcomp                                   \n",
       "381  eSDM                                          \n",
       "383  NaiveBayes                                    \n",
       "384  SDM-teaching                                  \n",
       "385  SDMproject                                    \n",
       "386  Composition                                   \n",
       "389  Statistical_Data_Mining-Final_Project         \n",
       "390  244_SMLW                                      \n",
       "391  PSM-Model                                     \n",
       "392  Light-Pollution-Project                       \n",
       "393  SDM-Final-Project                             \n",
       "394  SDMclass2018                                  \n",
       "395  SDMexoticPlants                               \n",
       "396  Makrooekologie                                \n",
       "397  sdmhw5                                        \n",
       "398  SDMProjectA                                   \n",
       "399  ECOP-VLab-SDM                                 \n",
       "400  STA546_SDM2                                   \n",
       "401  SDMDataAnalyticsProject                       \n",
       "402  landcover-sdms                                \n",
       "404  things                                        \n",
       "405  SDMPlay                                       \n",
       "406  SDMMetrics                                    \n",
       "407  sdmstack                                      \n",
       "408  SDMbase                                       \n",
       "409  SDMXWrappers                                  \n",
       "410  SDMstarter                                    \n",
       "411  SapienDistributionModel                       \n",
       "413  sdmhr                                         \n",
       "415  sdmflow                                       \n",
       "419  sdmanalysis                                   \n",
       "420  sdmid                                         \n",
       "422  RobiniaSDM                                    \n",
       "424  tapir_sdms                                    \n",
       "425  BioticSDMs                                    \n",
       "\n",
       "                                                                                                                  description  \\\n",
       "192  Search && Download Music from multi-platform                                                                               \n",
       "193  Menpo's 2D deformable modelling toolkit (AAMs/CLMs/SDMs)                                                                   \n",
       "194  Landmark with Regressition in Python （LBF(3000fps), ESR and SDM）                                                           \n",
       "195  Python interface to SDMX                                                                                                   \n",
       "197  Downloader SDM files via SDX file (DreamSpark) on Linux                                                                    \n",
       "199  Accelerated Attributed Network Embedding, SDM 2017                                                                         \n",
       "200  Python implementation of nonparametric nearest-neighbor-based estimators for divergences between distributions.            \n",
       "201  Spatial Data Modeler 5 for ArcGis pro                                                                                      \n",
       "202  Transportation Mode Detection For HTC                                                                                      \n",
       "204  Updated May 19, 2016                                                                                                       \n",
       "205  Updated Jan 9, 2019                                                                                                        \n",
       "206  Playing with implementing SDMP                                                                                             \n",
       "207  Docs for the GEN project.                                                                                                  \n",
       "208  Updated Apr 23, 2018                                                                                                       \n",
       "209  Updated Jun 11, 2017                                                                                                       \n",
       "210  IAMC data template implemented in SDMX                                                                                     \n",
       "211  Code and data for Deep Co-Clustering (SDM'19)                                                                              \n",
       "212  Updated Mar 13, 2019                                                                                                       \n",
       "213  Updated Mar 13, 2019                                                                                                       \n",
       "214  Updated Oct 30, 2015                                                                                                       \n",
       "215  Python Raspberry pi SDMCD16 control                                                                                        \n",
       "216  sdmx in python                                                                                                             \n",
       "217  An early Python implementation of sdmp/signed-yaml                                                                         \n",
       "218  Updated Feb 28, 2018                                                                                                       \n",
       "219  Problems and Solutions for Project Euler                                                                                   \n",
       "220  Using Sparse Distributed Memory for word segmentation. See if word boundaries can be discovered in highly-dimensional…     \n",
       "221  Network analysis for semantic data mining                                                                                  \n",
       "222  Updated May 6, 2019                                                                                                        \n",
       "223  Toolbox and scripts used to produce environmental variables for SDM regional project                                       \n",
       "224  Updated Nov 22, 2016                                                                                                       \n",
       "225  Updated Mar 13, 2019                                                                                                       \n",
       "226  Updated Feb 14, 2019                                                                                                       \n",
       "228  As part of Sequential Decision Making course, this class project aims to leverage deep learning to solve the Sub-T ch…     \n",
       "229  Example of the use of the single-decoupling method setup workflow                                                          \n",
       "230  this my project for PIMNAS 2018                                                                                            \n",
       "231  Tool to convert SDMX outputs to V4 format                                                                                  \n",
       "232  Updated Sep 29, 2017                                                                                                       \n",
       "233  First mini project for Semantic Data Management on Property graphs using Neo4j, UPC, BDMA                                  \n",
       "234  A python program to convert CTM files (usually generated by Kaldi) into an EMU SDMS database.                              \n",
       "235  Code and paper for an article presented at SDM'14 (and extended journal version in preparation)                            \n",
       "236  script collection using sdmpy to visualize peak amplitudes and switched power for VLA SDM datasets                         \n",
       "237  Code of paper \"Accelerated Experimental Design for Pairwise Comparisons\", by Guo et al., SDM 2019.                         \n",
       "238  Code for the paper \"An LSTM approach to Patent Classification based on Fixed Hierarchy Vectors\" to be presented in SDM …   \n",
       "239  Spatial Audio Python Package                                                                                               \n",
       "240  Binary Features）方法以及SDM（Supervised Decent Method）方法。但是，上述方法也存在一定的问题，例如，基于提取到的局部图像特征来定位时很难找到具有全局信息的关键点。基于点分布模型的方法和基于形…      \n",
       "241  Use Supervised Descent Method for 3D pose estimation.                                                                      \n",
       "242  SDM630-Modbus 3P4W Multifunction Power Analyser WithRS485 Port modbus RTU for the domoticz plugin                          \n",
       "243  example implementation of a epsilon greedy bandit algorithm for algorithm workshop                                         \n",
       "244  Read SDMX XML files using Python                                                                                           \n",
       "245  Implementation of sentiment analysis algorithms using labMT word set - and extending the set using Bayes classifier.       \n",
       "246  Python classes for reading VLA/ALMA Science Data Model (SDM)                                                               \n",
       "247  Widukind connector for PandaSDMX                                                                                           \n",
       "248  Simple script to get values from Eastron energy meter to Carbon                                                            \n",
       "249  Supplementary material for SDM paper \"LSCP: Locally Selective Combination in Parallel Outlier Ensembles\"                   \n",
       "250  Python toolkit for reading the radio astronomy data format known as the Science Data Model (SDM)                           \n",
       "251  Updated Dec 14, 2017                                                                                                       \n",
       "252  Specificity Determining Matrices of Co-Evolving Interaction Interfaces                                                     \n",
       "253  software defined efficient multicast                                                                                       \n",
       "254  Development environment for tools related to SDM prep and processing.                                                      \n",
       "256  read Eastron SDM 120/230 measurements via ModBus and store them to InfluxDB                                                \n",
       "257  Smokey feedback-based realtime machine learning thing.                                                                     \n",
       "258  SDM RDFizer                                                                                                                \n",
       "259  SotongDJ mplayer control script set for Kindle Touch Launcher                                                              \n",
       "260  Python wrapper for xSDM utility                                                                                            \n",
       "261  SDM630 - modbus                                                                                                            \n",
       "262  Collection of small scripts & tools                                                                                        \n",
       "263  Python library for reading data from SDM630 with MQTT interface                                                            \n",
       "264  Updated Apr 29, 2017                                                                                                       \n",
       "265  Towards scalable SDM (Hedwig update)                                                                                       \n",
       "266  MIT license                                                                                                                \n",
       "267  EASTON SDM630                                                                                                              \n",
       "268  Code related to SDM 2018 Paper \"Graph sketching-based Space-efficient Data Clustering\"                                     \n",
       "269  Tools for monitoring an SDM630 3 phase meter and reporting output to PVoutput                                              \n",
       "270  Code and experiments for \"A confidence-based approach for balancing fairness and accuracy\"                                 \n",
       "271  A variety of different display based OpenCV programs for the SD Mines UAS and Robotics Labs.                               \n",
       "272  GPL-3.0 license                                                                                                            \n",
       "273  A python script that reads a google form results and parses the results into a CSV file. The data compatible with Dev…     \n",
       "274  General python scripts for SDM projects                                                                                    \n",
       "275  NTU CSIE 2018 Fall, Special Topics/Directions on Machine Leaning (SDML)                                                    \n",
       "276  QGIS Plugin to query SDMX data cubes served as WFS feature types                                                           \n",
       "277  Basic dream-blog                                                                                                           \n",
       "278  Senior Design Mini Project                                                                                                 \n",
       "281  Simple Daemon Manager                                                                                                      \n",
       "282  Updated Dec 18, 2016                                                                                                       \n",
       "283  Encryption algorithm based on circle geometry.                                                                             \n",
       "284  Python code prototyping ANT+ SDM Sensor                                                                                    \n",
       "285  Updated Apr 1, 2019                                                                                                        \n",
       "286  Converting sdg data to SDMX                                                                                                \n",
       "287  Final Project: ROB534 - Sequential Decision Making in Robotics                                                             \n",
       "288  Docker build files for scientific data management                                                                          \n",
       "289  Updated Nov 17, 2016                                                                                                       \n",
       "290  It's a course final project which is developed in a course name 'Software Develop Method' in NTUIM.                        \n",
       "291  GPL-3.0 license                                                                                                            \n",
       "292  Updated Apr 23, 2019                                                                                                       \n",
       "293  Updated Dec 15, 2012                                                                                                       \n",
       "294  Python scripts to convert SDMX files to CSV. Initially only for Statscan Census files.                                     \n",
       "295  WASH WoS Data Management and Data Analysis                                                                                 \n",
       "296  Updated Dec 12, 2017                                                                                                       \n",
       "298  UI for RJSDMX                                                                                                              \n",
       "300  modules for each component of an SDM workflow, for use with the zoon R package                                             \n",
       "301  Updated Feb 27, 2015                                                                                                       \n",
       "302  Updated Aug 16, 2017                                                                                                       \n",
       "304  Updated Mar 20, 2019                                                                                                       \n",
       "305  Updated Mar 19, 2018                                                                                                       \n",
       "307  Updated Jan 19, 2017                                                                                                       \n",
       "310  Updated Dec 29, 2018                                                                                                       \n",
       "311  Updated Oct 19, 2018                                                                                                       \n",
       "312  Updated Jul 11, 2016                                                                                                       \n",
       "314  Analysis of the Social Decision Making Algorithms                                                                          \n",
       "316  Updated Mar 19, 2019                                                                                                       \n",
       "318  Updated Sep 4, 2015                                                                                                        \n",
       "320  Updated Apr 18, 2016                                                                                                       \n",
       "321  Updated Oct 20, 2017                                                                                                       \n",
       "322  Updated Apr 11, 2019                                                                                                       \n",
       "323  results from my biofungicide trials that I ran in the Salinas/Pajaro valley for spinach downy mildew in 2013-2016.         \n",
       "324  Statistics, Data Mining and Apprentissage Exercises                                                                        \n",
       "325  R package with useful functions for working with SDMs and Natural Heritage data                                            \n",
       "327  Updated Sep 4, 2015                                                                                                        \n",
       "328  Proyecto de modelación de distribución potencial de Dalbergias                                                             \n",
       "330  Updated Aug 27, 2018                                                                                                       \n",
       "331  R script to automate generation of SDM files                                                                               \n",
       "332  Package to calculate Minimum Background Area for SDMs                                                                      \n",
       "333  code used in manuscript - Potential impacts of a future persistent El Nino or La Nina on three subspecies of Australi…     \n",
       "334  Updated May 1, 2019                                                                                                        \n",
       "335  Project modelling the change of six species within the Severn Estuary and the impact of the Severn Barrage and Cardif…     \n",
       "337  Use local ecological knowledge to quantitatively evalaute SDM outputs                                                      \n",
       "338  Updated Mar 28, 2017                                                                                                       \n",
       "340  Introductory lesson to generate range maps for butterfly-host plant interactions and predict distributional shifts us…     \n",
       "341  A simple example of how to use drake to create a pipeline for SDMs.                                                        \n",
       "343  201603_Ecology                                                                                                             \n",
       "344  Updated Jul 20, 2015                                                                                                       \n",
       "345  Code for generating virtual species and testing SDM performance                                                            \n",
       "347  AGPL-3.0 license                                                                                                           \n",
       "349  Statistical Decision Methods at SGH, summer semester 2017/18                                                               \n",
       "350  Updated Apr 21, 2016                                                                                                       \n",
       "352  General scripts necessary for steps during the SDM approach                                                                \n",
       "353  Model fitting and evaluation for dynamic ensemble blue whale distribution model                                            \n",
       "354  Updated May 10, 2019                                                                                                       \n",
       "355  The Repository Models FIFA 19 Players and gives an insight on what traits in a Player is important in judging the Ove…     \n",
       "356  A version of Nicholas J Lyon's modeling thesis chapter (SDMs) that is ready to be made publicly available post-public…     \n",
       "357  A basic test fixture for testing the impact of different absence data approaches in seegSDM                                \n",
       "358  Updated Apr 25, 2019                                                                                                       \n",
       "360  Various bbs related modeling I'm doing. One of them, a meta analysis of bbs SDM's, is now in rep bbs-SDM                   \n",
       "361  Spatially explicit version of a State and Transition Model with forest management                                          \n",
       "362  Working scripts for creating SDMs and decomposing predictors to assess variable importance under climate change            \n",
       "363  Compilation of ex situ collections, and in situ occurrence points; creation of SDMs                                        \n",
       "364  Repository for the SDM simulation model associated with Yanco et al. 2018                                                  \n",
       "365  INFO522 proejct repository cloned from https://github.com/jcoliver/ebutterfly-sdm                                          \n",
       "366  Scripts to map phylogenetic diversity from IUCN maps or SDM´s                                                              \n",
       "367  Methods and code for assembling the State Lands PROs: polygon processing, SDMs, hypergrid, polygon/raster products         \n",
       "368  Shiny App for the analytical analysis of the State and Transition Model with forest management                             \n",
       "369  Thesis for GIST master’s at USC-SSCI. Study compares tuned versus default MaxEnt models at two different covariate re…     \n",
       "371  MIT license                                                                                                                \n",
       "372  An R package for spatial and spatiotemporal GLMMs with TMB                                                                 \n",
       "377  Tools for working with Maxent in R                                                                                         \n",
       "378  Exploring the potential to incorporate physiological priors in species distribution models                                 \n",
       "380  Scripts used in Maguire et al....                                                                                          \n",
       "381  A user-friendly spatial tool with a web-based interface that allows users to import SDM layers and create and explore…     \n",
       "383  SDM                                                                                                                        \n",
       "384  Teach SDM                                                                                                                  \n",
       "385  SDM project                                                                                                                \n",
       "386  SDMs spatial SDM composition using an automatic R Script                                                                   \n",
       "389  SDM final Project                                                                                                          \n",
       "390  Intro, SDM, Island Variables                                                                                               \n",
       "391  PSM Modeling for SDM                                                                                                       \n",
       "392  SDM Final Project                                                                                                          \n",
       "393  NCKU SDM Final Project                                                                                                     \n",
       "394  Updated Mar 19, 2018                                                                                                       \n",
       "395  code used to create outputs for weedfutures.net and article: Next-generation invaders? Hotspots for naturalised sleep…     \n",
       "396  Kurs Makroökologie, erstellen von SDM                                                                                      \n",
       "397  Naive Bayes, Recursive partition tree and gradient boosting model                                                          \n",
       "398  Updated Apr 8, 2019                                                                                                        \n",
       "399  ECOPOTENTIAL IRIS SDM model/service                                                                                        \n",
       "400  Contains code for assignments in UB STA546 Course                                                                          \n",
       "401  Updated Jan 12, 2019                                                                                                       \n",
       "402  Scripts for sdm post processing using landcover                                                                            \n",
       "404  Miscellaneous R functions - mainly spatial and SDM                                                                         \n",
       "405  Updated Oct 10, 2017                                                                                                       \n",
       "406  Updated Jun 7, 2018                                                                                                        \n",
       "407  Updated Jul 20, 2016                                                                                                       \n",
       "408  GPL-3.0 license                                                                                                            \n",
       "409  A collection of wrappers on top of RJSDMX package                                                                          \n",
       "410  Updated Apr 8, 2019                                                                                                        \n",
       "411  Debiasing of SDM using Social Media Data                                                                                   \n",
       "413  Updated Sep 12, 2014                                                                                                       \n",
       "415  Updated Feb 8, 2019                                                                                                        \n",
       "419  Updated Sep 4, 2015                                                                                                        \n",
       "420  Repository for blog code                                                                                                   \n",
       "422  Species distribution modeling of the nitrogen-fixing tree black locust                                                     \n",
       "424  Dissertation Research                                                                                                      \n",
       "425  Updated Mar 3, 2016                                                                                                        \n",
       "\n",
       "                                                                       url  \n",
       "192  https://github.com/pcdack/SDMUSIC                                      \n",
       "193  https://github.com/menpo/menpofit                                      \n",
       "194  https://github.com/FacialLandmark/landmark_py                          \n",
       "195  https://github.com/dr-leo/pandaSDMX                                    \n",
       "197  https://github.com/RadekSimkanic/Linux-SDM-Downloader                  \n",
       "199  https://github.com/xhuang31/AANE_Python                                \n",
       "200  https://github.com/dougalsutherland/py-sdm                             \n",
       "201  https://github.com/gtkfi/ArcSDM                                        \n",
       "202  https://github.com/BaiiYuan/SDML-Final                                 \n",
       "204  https://github.com/caseyjlaw/sdmpy-old                                 \n",
       "205  https://github.com/wheatdog/sdml_final                                 \n",
       "206  https://github.com/jmhobbs/sdmp-python                                 \n",
       "207  https://github.com/vinay87/gen_sdm                                     \n",
       "208  https://github.com/tawjaw/sparkSDM                                     \n",
       "209  https://github.com/YunsikPark/Django-sdModels                          \n",
       "210  https://github.com/khaeru/iamc-sdmx-demo                               \n",
       "211  https://github.com/DerronXu/Deep-Co-Clustering                         \n",
       "212  https://github.com/haydarai/sdm-lab-1                                  \n",
       "213  https://github.com/PinarTurkyilmaz/SDM-Lab1                            \n",
       "214  https://github.com/jaeindia/SDMA-Community-Detection                   \n",
       "215  https://github.com/georgeredinger/pipySDM-CD16                         \n",
       "216  https://github.com/seraekim/srkim-py-sdmx                              \n",
       "217  https://github.com/jmhobbs/sdmp-signed-yaml-python                     \n",
       "218  https://github.com/VANatHeritage/SDM-PresencePreProc                   \n",
       "219  https://github.com/sdmx/project-euler-sdmx                             \n",
       "220  https://github.com/yczeng/sdm-word-segmentation                        \n",
       "221  https://github.com/JanKralj/cf_netSDM                                  \n",
       "222  https://github.com/haydarai/sdm-lab-3                                  \n",
       "223  https://github.com/VANatHeritage/SDM-ArcToolsEV                        \n",
       "224  https://github.com/charlienewey/sdmr-paper-code                        \n",
       "225  https://github.com/hugrave-upc/SDM_Lab1                                \n",
       "226  https://github.com/YiHerngOng/ROB534_SDM                               \n",
       "228  https://github.com/abhiagni11/Deep-Learning-SDM                        \n",
       "229  https://github.com/egallicc/openmm_sdm_workflow                        \n",
       "230  https://github.com/imamhakim/project-SDM-Soil-Data-Monitoring-         \n",
       "231  https://github.com/ONS-OpenData/cmd-sdmx-to-v4                         \n",
       "232  https://github.com/jm921106/sdm-python-raspiSerialLib                  \n",
       "233  https://github.com/ankushs92/upc-sdm-property-graphs-lab1              \n",
       "234  https://github.com/danijel3/CTMtoEMU                                   \n",
       "235  https://github.com/rionda/truefis                                      \n",
       "236  https://github.com/fschinzel/rfisweep                                  \n",
       "237  https://github.com/neu-spiral/AcceleratedExperimentalDesign            \n",
       "238  https://github.com/marawanokasha/fhv_paper_code                        \n",
       "239  https://github.com/chris-hld/spaudiopy                                 \n",
       "240  https://github.com/1365843167/-                                        \n",
       "241  https://github.com/ChrisYang/SDM3Dpose                                 \n",
       "242  https://github.com/MFxMF/SDM630-Modbus                                 \n",
       "243  https://github.com/brentpayne/sdml-bandit-example                      \n",
       "244  https://github.com/mwilliamson/sdmx.py                                 \n",
       "245  https://github.com/BogdanSorlea/SentimentAnalysisSDM                   \n",
       "246  https://github.com/demorest/sdmpy                                      \n",
       "247  https://github.com/Widukind/widukind.py.pandaSDMX                      \n",
       "248  https://github.com/ranzhe/sdm220-carbon                                \n",
       "249  https://github.com/yzhao062/LSCP                                       \n",
       "250  https://github.com/caseyjlaw/sdmreader                                 \n",
       "251  https://github.com/victor-gonzalez-navarro/SDMS                        \n",
       "252  https://github.com/Bowloforanges/SDMCEII                               \n",
       "253  https://github.com/prithviraj6116/sdmc                                 \n",
       "254  https://github.com/PNHP/SDM_Tools                                      \n",
       "256  https://github.com/lesinigo/sdm2influx                                 \n",
       "257  https://github.com/Charcoal-SE/sdml                                    \n",
       "258  https://github.com/SDM-TIB/TIB-RDFizer                                 \n",
       "259  https://github.com/SotongDJ/sdmplayer-kt                               \n",
       "260  https://github.com/exander77/xSDM-python-wrapper                       \n",
       "261  https://github.com/coolex/sdm630                                       \n",
       "262  https://github.com/SDBagel/SDMiniTools                                 \n",
       "263  https://github.com/xtompok/SDM630                                      \n",
       "264  https://github.com/binderchri/sdm2                                     \n",
       "265  https://github.com/SkBlaz/hedwig3                                      \n",
       "266  https://github.com/develope126/SDMA2                                   \n",
       "267  https://github.com/ticapix/home-assistant_sdm630                       \n",
       "268  https://github.com/annemorvan/DBMSTClu                                 \n",
       "269  https://github.com/ruddj/sdm630-report                                 \n",
       "270  https://github.com/j2kun/fkl-SDM16                                     \n",
       "271  https://github.com/PhilTKamp/SDMinesLabDisplay                         \n",
       "272  https://github.com/nickma82/sdm220_eastron_wattmeter                   \n",
       "273  https://github.com/pemby/SDMAdressBookCreator                          \n",
       "274  https://github.com/VANatHeritage/py-SDM                                \n",
       "275  https://github.com/victai/SDML                                         \n",
       "276  https://github.com/lmorandini/SDMXPlugin                               \n",
       "277  https://github.com/Raveline/sdmb                                       \n",
       "278  https://github.com/thakkarV/SDMini                                     \n",
       "281  https://github.com/avl93/sdmgr                                         \n",
       "282  https://github.com/hazemsoliman115/SDMA                                \n",
       "283  https://github.com/Frajerzycki/SDMPPO                                  \n",
       "284  https://github.com/jonathanpandres/Python-Treadmill-Prototype          \n",
       "285  https://github.com/workingravi/csv_sdmx                                \n",
       "286  https://github.com/ONSdigital/sdg-sdmx                                 \n",
       "287  https://github.com/dietriro/sdm_project                                \n",
       "288  https://github.com/suchyta1/sdm-docker                                 \n",
       "289  https://github.com/miguelgamendes/SDM_CPAB                             \n",
       "290  https://github.com/weichih-c/sdm-project                               \n",
       "291  https://github.com/dbaldanza/sdmpv_bot                                 \n",
       "292  https://github.com/TNWOG/PICMath_SDMA                                  \n",
       "293  https://github.com/berkeley-cocosci/bayes-sdm                          \n",
       "294  https://github.com/plot-and-scatter/sdmx-converter                     \n",
       "295  https://github.com/udaraz77/WASHWoSDMDA                                \n",
       "296  https://github.com/mike840609/SDM_4                                    \n",
       "298  https://github.com/bowerth/sdmxBrowser                                 \n",
       "300  https://github.com/zoonproject/modules                                 \n",
       "301  https://github.com/adammwilson/hSDM_Tutorial                           \n",
       "302  https://github.com/oharar/PointedSDMs                                  \n",
       "304  https://github.com/saltpawpaw/QTP_SDM                                  \n",
       "305  https://github.com/mmsmith9/SDM_class                                  \n",
       "307  https://github.com/gallington/Mongolia_SDM                             \n",
       "310  https://github.com/freyagenesis/SDM-Project                            \n",
       "311  https://github.com/RushingLab/BBS.SDM                                  \n",
       "312  https://github.com/crollinson/Establishment_SDMs                       \n",
       "314  https://github.com/rpitrust/sdm_analysis                               \n",
       "316  https://github.com/KIT-IfGG/Malaria_SDM                                \n",
       "318  https://github.com/howl-anderson/sdmengine.common                      \n",
       "320  https://github.com/KristinaRiemer/SDM_traits                           \n",
       "321  https://github.com/sdtaylor/bbs-SDM                                    \n",
       "322  https://github.com/aminorberg/SDM-comparison                           \n",
       "323  https://github.com/robchoudhury/sdm_biofungicide                       \n",
       "324  https://github.com/KK-Projects/SDMA_Exercises                          \n",
       "325  https://github.com/VANatHeritage/nhSDM                                 \n",
       "327  https://github.com/howl-anderson/sdmengine.dismo                       \n",
       "328  https://github.com/fpardourrutia/sdm_dalbergias                        \n",
       "330  https://github.com/bobmuscarella/AMF_SDMs                              \n",
       "331  https://github.com/Joe-Hilgard/BV-SDM_automator                        \n",
       "332  https://github.com/xavi-rp/MinBAR                                      \n",
       "333  https://github.com/daisyduursma/SDM_ENSO_Butterflies                   \n",
       "334  https://github.com/KatieMCook/SDM_course_git                           \n",
       "335  https://github.com/laxmack21/Severn-Estuary-SDMS                       \n",
       "337  https://github.com/torrem/Local-Ecological-knowledge-Model-evaluation  \n",
       "338  https://github.com/lukasbaumbach/beech-fir-sdm                         \n",
       "340  https://github.com/jcoliver/biodiversity-sdm-lesson                    \n",
       "341  https://github.com/jequihua/simple_sdm_pipeline                        \n",
       "343  https://github.com/kanji14134/2016_Ecology_SDM                         \n",
       "344  https://github.com/tlmizerek/coral_traits_SDM                          \n",
       "345  https://github.com/bellland/SDM.Virtual.Species_Bell.Schlaepfer        \n",
       "347  https://github.com/jallen73/SDM-batch-scripts                          \n",
       "349  https://github.com/nosarzewski/SGH_SDM_18                              \n",
       "350  https://github.com/dschlaep/sdm.nlu2016ss                              \n",
       "352  https://github.com/Rilquer/speciesdistributionmodeling                 \n",
       "353  https://github.com/briana-abrahms/DynamicEnsembleSDM                   \n",
       "354  https://github.com/HMB3/Plant_SDM_code                                 \n",
       "355  https://github.com/aashishkakar/SDM-Model-FIFA-19                      \n",
       "356  https://github.com/NJLyon-Projects/Lyon.MS.Thesis-SDM.Project          \n",
       "357  https://github.com/fshearer/seegSDM-bias-testing-fixture               \n",
       "358  https://github.com/christianhof/BioScen1.5_SDM                         \n",
       "360  https://github.com/sdtaylor/bbs                                        \n",
       "361  https://github.com/willvieira/STManaged                                \n",
       "362  https://github.com/naiamh/Predictor_decomposition                      \n",
       "363  https://github.com/esbeckman/oak_gap_analysis                          \n",
       "364  https://github.com/syanco/checkyourself                                \n",
       "365  https://github.com/kensaku-okada/INFO522Project                        \n",
       "366  https://github.com/lgchumboldt/Mapping_phylogenetic_diversity          \n",
       "367  https://github.com/NYNatHeritage/PROs                                  \n",
       "368  https://github.com/willvieira/shiny_STM-managed                        \n",
       "369  https://github.com/CassKal/GIST_Thesis                                 \n",
       "371  https://github.com/jcoliver/ebutterfly-sdm                             \n",
       "372  https://github.com/pbs-assess/sdmTMB                                   \n",
       "377  https://github.com/johnbaums/rmaxent                                   \n",
       "378  https://github.com/HuckleyLab/SDMpriors                                \n",
       "380  https://github.com/fitzLab-AL/SDM-CLMcomp                              \n",
       "381  https://github.com/smwoodman/eSDM                                      \n",
       "383  https://github.com/swapnilawasthi/NaiveBayes                           \n",
       "384  https://github.com/madanksuwal/SDM-teaching                            \n",
       "385  https://github.com/swapnilawasthi/SDMproject                           \n",
       "386  https://github.com/ccsosa/Composition                                  \n",
       "389  https://github.com/gauravjetley/Statistical_Data_Mining-Final_Project  \n",
       "390  https://github.com/alyssa-a-winchell/244_SMLW                          \n",
       "391  https://github.com/mcolvin/PSM-Model                                   \n",
       "392  https://github.com/sonia3187/Light-Pollution-Project                   \n",
       "393  https://github.com/maciao/SDM-Final-Project                            \n",
       "394  https://github.com/kthompson25/SDMclass2018                            \n",
       "395  https://github.com/daisyduursma/SDMexoticPlants                        \n",
       "396  https://github.com/debroize/Makrooekologie                             \n",
       "397  https://github.com/swapnilawasthi/sdmhw5                               \n",
       "398  https://github.com/hayeshursh/SDMProjectA                              \n",
       "399  https://github.com/joaofgoncalves/ECOP-VLab-SDM                        \n",
       "400  https://github.com/varadtupe/STA546_SDM2                               \n",
       "401  https://github.com/rahilshaik/SDMDataAnalyticsProject                  \n",
       "402  https://github.com/LBAB-Humboldt/landcover-sdms                        \n",
       "404  https://github.com/johnbaums/things                                    \n",
       "405  https://github.com/charleneguillaumot/SDMPlay                          \n",
       "406  https://github.com/Farewe/SDMMetrics                                   \n",
       "407  https://github.com/davharris/sdmstack                                  \n",
       "408  https://github.com/RWunderlich/SDMbase                                 \n",
       "409  https://github.com/jcizel/SDMXWrappers                                 \n",
       "410  https://github.com/Lin-Jing-R/SDMstarter                               \n",
       "411  https://github.com/guyawn/SapienDistributionModel                      \n",
       "413  https://github.com/grgurev/sdmhr                                       \n",
       "415  https://github.com/juoe/sdmflow                                        \n",
       "419  https://github.com/howl-anderson/sdmanalysis                           \n",
       "420  https://github.com/daniel-rodak/sdmid                                  \n",
       "422  https://github.com/JeffreyMinucci/RobiniaSDM                           \n",
       "424  https://github.com/codyschank/tapir_sdms                               \n",
       "425  https://github.com/katieleach/BioticSDMs                               "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# This is a manual step\n",
    "df[df.keywords == 'SDM'][['name', 'description', 'url']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "names_to_keep = ['nhSDM', 'MinBAR', 'SDM_ENSO_Butterflies',\n",
    "                 'SDM_course_git', 'Severn-Estuary-SDMS', 'Local-Ecological-knowledge-Model-evaluation',\n",
    "                 'biodiversity-sdm-lesson', '2016_Ecology_SDM', 'coral_traits_SDM',\n",
    "                 'SDM.Virtual.Species_Bell.Schlaepfer', 'speciesdistributionmodeling', 'DynamicEnsembleSDM',\n",
    "                 'Plant_SDM_code', 'Lyon.MS.Thesis-SDM.Project', 'BioScen1.5_SDM', 'bbs', 'STManaged',\n",
    "                 'Predictor_decomposition', 'oak_gap_analysis', 'Mapping_phylogenetic_diversity',\n",
    "                 'PROs', 'shiny_STM-managed', 'GIST_Thesis', 'ebutterfly-sdm', 'sdmTMB', 'rmaxent',\n",
    "                 'SDMpriors', 'eSDM', 'SDM-CLMcomp', 'SDM-teaching', 'Composition', '244_SMLW',\n",
    "                 'Light-Pollution-Project', 'SDMexoticPlants', 'Makrooekologie', 'SDMProjectA', 'STA546_SDM2',\n",
    "                 'SDMDataAnalyticsProject', 'RobiniaSDM', 'things', 'BioticSDMs', 'tapir_sdms'] \n",
    "\n",
    "# must be packages or substantial repos about SDM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "625\n",
      "476\n"
     ]
    }
   ],
   "source": [
    "print(len(df))\n",
    "df_sdm = df[(df.name.isin(names_to_keep)) & (df.keywords == 'SDM')]\n",
    "df = df[df.keywords != 'SDM'].append(df_sdm)\n",
    "print(len(df))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check niche and remove irrelevant packages\n",
    "Treatment: remove"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>CENFA</td>\n",
       "      <td>Climate and Ecological Niche Factor Analysis</td>\n",
       "      <td>https://cran.r-project.org/web/packages/CENFA/index.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ENiRG</td>\n",
       "      <td>Ecological Niche in R and GRASS</td>\n",
       "      <td>https://cran.r-project.org/web/packages/ENiRG/index.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>ENMeval</td>\n",
       "      <td>Automated Runs and Evaluations of Ecological Niche Models</td>\n",
       "      <td>https://cran.r-project.org/web/packages/ENMeval/index.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>EnvNicheR</td>\n",
       "      <td>Niche Estimation</td>\n",
       "      <td>https://cran.r-project.org/web/packages/EnvNicheR/index.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>MaxentVariableSelection</td>\n",
       "      <td>Selecting the Best Set of Relevant Environmental Variables along with the Optimal Regularization Multiplier for Maxent Niche Modeling</td>\n",
       "      <td>https://cran.r-project.org/web/packages/MaxentVariableSelection/index.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>nicheROVER</td>\n",
       "      <td>(Niche) (R)egion and Niche (Over)lap Metrics for Multidimensional Ecological Niches</td>\n",
       "      <td>https://cran.r-project.org/web/packages/nicheROVER/index.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>phyloclim</td>\n",
       "      <td>Integrating Phylogenetics and Climatic Niche Modeling</td>\n",
       "      <td>https://cran.r-project.org/web/packages/phyloclim/index.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>rKIN</td>\n",
       "      <td>(Kernel) Isotope Niche Estimation</td>\n",
       "      <td>https://cran.r-project.org/web/packages/rKIN/index.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>xpost-bot</td>\n",
       "      <td>Reddit bot to scan and repost submissions of interest to niche subreddits</td>\n",
       "      <td>https://github.com/git2samus/xpost-bot</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>fynat</td>\n",
       "      <td>Find Your Niche At Tech</td>\n",
       "      <td>https://github.com/dadams22/fynat</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ambitus</td>\n",
       "      <td>Ancestral niche reconstruction and associated tools.</td>\n",
       "      <td>https://github.com/ryanafolk/ambitus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>opalescence</td>\n",
       "      <td>Niche-purpose discord moderation utility bot</td>\n",
       "      <td>https://github.com/NiteAx/opalescence</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>niche_vlaanderen</td>\n",
       "      <td>Python package to run the NICHE Vlaanderen model</td>\n",
       "      <td>https://github.com/inbo/niche_vlaanderen</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NicheLife</td>\n",
       "      <td>A map visualization tool to find your ideal neighborhoods in Manhattan</td>\n",
       "      <td>https://github.com/priya-vijay/NicheLife</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>nicheLang</td>\n",
       "      <td>Updated May 22, 2018</td>\n",
       "      <td>https://github.com/mattHoyDev/nicheLang</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>language-niche-research</td>\n",
       "      <td>Scientific repository for fiddling around with linguistic data</td>\n",
       "      <td>https://github.com/RichardLitt/language-niche-research</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NichePSO</td>\n",
       "      <td>Updated Aug 22, 2018</td>\n",
       "      <td>https://github.com/bryaneduardo24/NichePSO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>niche</td>\n",
       "      <td>MIT license</td>\n",
       "      <td>https://github.com/rwsharp/niche</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>nicheCrawler</td>\n",
       "      <td>nicheCrawler is a bot that crawls through niche.com and returns all universities from desired state.</td>\n",
       "      <td>https://github.com/onthir/nicheCrawler</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NICHE</td>\n",
       "      <td>Code for UNICEF NICHE project.</td>\n",
       "      <td>https://github.com/alison-campion/NICHE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NicheReviewScraperPY</td>\n",
       "      <td>Updated Oct 22, 2015</td>\n",
       "      <td>https://github.com/SazanDauti/NicheReviewScraperPY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>nicheRecommender</td>\n",
       "      <td>Provides niche, lesser-known places for people to visit, explore, eat at, etc.</td>\n",
       "      <td>https://github.com/Qwertazertl0/nicheRecommender</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Nichehire</td>\n",
       "      <td>Updated Dec 6, 2017</td>\n",
       "      <td>https://github.com/saikeerthana19/Nichehire</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>nicheapps</td>\n",
       "      <td>nicheapps is a platform to showcase your apps to the users , post a demo , get feedback</td>\n",
       "      <td>https://github.com/addyrookie/nicheapps</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>niches_finder</td>\n",
       "      <td>Updated Mar 21, 2019</td>\n",
       "      <td>https://github.com/lukadamczyk/niches_finder</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Coding-niche</td>\n",
       "      <td>Updated Oct 14, 2018</td>\n",
       "      <td>https://github.com/JayatiGoyal/Coding-niche</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>MyNiche</td>\n",
       "      <td>Website built to allow clubs to share notifications in real time with school students. Manages club accounts and stud…</td>\n",
       "      <td>https://github.com/aiswaryasankar/MyNiche</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>niche_modeling</td>\n",
       "      <td>Updated Sep 3, 2014</td>\n",
       "      <td>https://github.com/tomquisel/niche_modeling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>nndocker</td>\n",
       "      <td>Neural Niche Docker Config</td>\n",
       "      <td>https://github.com/grahamannett/nndocker</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>app-niche</td>\n",
       "      <td>API RESTful to find your app niche in Flask</td>\n",
       "      <td>https://github.com/costular/app-niche</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>niche-scraping</td>\n",
       "      <td>Updated Mar 30, 2019</td>\n",
       "      <td>https://github.com/rydena021/niche-scraping</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>niche-construction</td>\n",
       "      <td>Studies on niche construction through agent based modeling.</td>\n",
       "      <td>https://github.com/CA-Lab/niche-construction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Serendipity</td>\n",
       "      <td>/r/serendipity is a meta-subreddit meant to broaden the perspective of its subscribers. It takes a popular entry from…</td>\n",
       "      <td>https://github.com/umbrae/Serendipity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>api.niche.fm</td>\n",
       "      <td>Updated Mar 17, 2018</td>\n",
       "      <td>https://github.com/hery/api.niche.fm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>powers_niche_construction</td>\n",
       "      <td>Powers et al's niche construction model</td>\n",
       "      <td>https://github.com/greenape/powers_niche_construction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Temporal_niche_modeling</td>\n",
       "      <td>Updated Feb 14, 2018</td>\n",
       "      <td>https://github.com/gaodifan/Temporal_niche_modeling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Random-Niche-Generation</td>\n",
       "      <td>Randomly creates a sequence of niches within the full nuclear fuel cycle based on a dictionary of plausible connections.</td>\n",
       "      <td>https://github.com/aglenn2/Random-Niche-Generation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>lifeNicheTemp</td>\n",
       "      <td>lifeNicheTemp</td>\n",
       "      <td>https://github.com/Bouteloua/lifeNicheTemp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Niche-EnterpriseProject</td>\n",
       "      <td>Updated Apr 19, 2017</td>\n",
       "      <td>https://github.com/vinaykl/Niche-EnterpriseProject</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Niche-School-Scraper</td>\n",
       "      <td>A Python script that uses BeautifulSoup and requests to scrape the addresses from high schools on niche.com.</td>\n",
       "      <td>https://github.com/haanmiba/Niche-School-Scraper</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Instagram-Niche-Downloader</td>\n",
       "      <td>Download images content as per your niche.</td>\n",
       "      <td>https://github.com/ougrocks/Instagram-Niche-Downloader</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>netyell</td>\n",
       "      <td>sends formatted messages to symon netbrite signs from the command line. (yep, there's a niche!)</td>\n",
       "      <td>https://github.com/rsaxvc/netyell</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>Python-Scripts</td>\n",
       "      <td>Small python scripts for niche uses.</td>\n",
       "      <td>https://github.com/keshavd/Python-Scripts</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>DNAP</td>\n",
       "      <td>Release checker for niche vinyl labels</td>\n",
       "      <td>https://github.com/Tenchi2xh/DNAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>GA_Lab</td>\n",
       "      <td>Genetic algorithm niching functions lab work</td>\n",
       "      <td>https://github.com/watemerald/GA_Lab</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>hong_kong_waiting_time_reused_niches</td>\n",
       "      <td>Hong Kong Waiting Time for Allocation of Reused Niches</td>\n",
       "      <td>https://github.com/howawong/hong_kong_waiting_time_reused_niches</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Differential-Evolution</td>\n",
       "      <td>差分进化算法，支持小生境（niche）技术，支持多目标</td>\n",
       "      <td>https://github.com/sgdd66/Differential-Evolution</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>peutils</td>\n",
       "      <td>Binary Ninja plugin providing various niche utilities for working with PE binaries</td>\n",
       "      <td>https://github.com/404d/peutils</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>PCSDswag</td>\n",
       "      <td>A niche screen scraper for just one (1) website</td>\n",
       "      <td>https://github.com/macsj200/PCSDswag</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>NMMSO</td>\n",
       "      <td>Python Implementation of the Niching Migratory Multi-Swarm Optimizer</td>\n",
       "      <td>https://github.com/varchasgopalaswamy/NMMSO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>EntheoBot</td>\n",
       "      <td>Discord bot for a niche group of chatters.</td>\n",
       "      <td>https://github.com/DrewTChrist/EntheoBot</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>EasyList</td>\n",
       "      <td>A niche type of list keeper in python.</td>\n",
       "      <td>https://github.com/wesselmeijer/EasyList</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>instaBot</td>\n",
       "      <td>upgraded tool to follow an like users posts on your niche . u need to enter your username and password then your niche …</td>\n",
       "      <td>https://github.com/crhexn8belda/instaBot</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>moneyball_app</td>\n",
       "      <td>A webapp to track results of foosball games in financial institutions. It is a very niche market!</td>\n",
       "      <td>https://github.com/moneyball-dk/moneyball_app</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>opensim-stl-export</td>\n",
       "      <td>Want to export a model from OpenSim to STL? Wegotchufam.</td>\n",
       "      <td>https://github.com/prnthp/opensim-stl-export</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>CTCI</td>\n",
       "      <td>interview practice from CTCI, leetcode, hackerrank, and other niches of the web</td>\n",
       "      <td>https://github.com/jazzminewang/CTCI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>US-Universities-Scraper</td>\n",
       "      <td>Scraper that provides information on universities from various sites(ex. USNews, Niche)</td>\n",
       "      <td>https://github.com/Jenniferrrrrr/US-Universities-Scraper</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>blender-project-tools</td>\n",
       "      <td>Code &amp; tools made to assist with various niche Moo-Ack! Productions projects</td>\n",
       "      <td>https://github.com/TheDuckCow/blender-project-tools</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>MunichBFOR</td>\n",
       "      <td>Evolutionary Multi Niche Bacterial Foraging Algorithm for Multimodal Dynamic Optimizations and Neural Network Optimiz…</td>\n",
       "      <td>https://github.com/dananjayamahesh/MunichBFOR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>MusicAnalysisTools</td>\n",
       "      <td>music21 based tools for analysis. Often WIP and / or too niche for the main music21 repo</td>\n",
       "      <td>https://github.com/MarkGotham/MusicAnalysisTools</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>write_for_us</td>\n",
       "      <td>Script for finding niche websites via Google search that accept article contributions</td>\n",
       "      <td>https://github.com/davejonesbkk/write_for_us</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>multi-modal-ga</td>\n",
       "      <td>A multi-modal genetic algorithm which will use two niching methods.</td>\n",
       "      <td>https://github.com/serbanc94/multi-modal-ga</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>fuzzy_habitat_modelling</td>\n",
       "      <td>Fish habitat model and visualization of habitat niche using GRASS GIS r.fuzzy.system and R</td>\n",
       "      <td>https://github.com/jradinger/fuzzy_habitat_modelling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>Youtuber-Discovery</td>\n",
       "      <td>Single page web app for finding or researching youtubers in a given niche or area</td>\n",
       "      <td>https://github.com/afimbufe/Youtuber-Discovery</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>MAGA</td>\n",
       "      <td>A Multi-niche Adaptive Genetic Algorithm for biomarker discovery from omics data.</td>\n",
       "      <td>https://github.com/Greco-Lab/MAGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>SelfDestructBot</td>\n",
       "      <td>A niche moderator bot that removes posts after a specified amount of time.</td>\n",
       "      <td>https://github.com/diceroll123/SelfDestructBot</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>BATty</td>\n",
       "      <td>A Python powered serverless web app for understanding the current market value of niche used cars</td>\n",
       "      <td>https://github.com/dhicks43/BATty</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>WallpaperSort</td>\n",
       "      <td>A Python script to sort 1.6 and 1.7 wallpapers into separate folders. Kinda niche, to be honest.</td>\n",
       "      <td>https://github.com/KBKarma/WallpaperSort</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>CommNetAntennasExtension</td>\n",
       "      <td>Add 6 new antennas for CommNet, filling a niche in the transmission ranges: direct 5M, relay 32M, relays 7G and 25G, …</td>\n",
       "      <td>https://github.com/yalov/CommNetAntennasExtension</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>EccentricRecommender</td>\n",
       "      <td>Using a new Metric called Eccentricity this Recommender System aims on improving the variations in modern Recommendat…</td>\n",
       "      <td>https://github.com/SandroLuck/EccentricRecommender</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>image_checker</td>\n",
       "      <td>Used for checking one locations files against another locations files to find the difference (if one exists), extremely</td>\n",
       "      <td>https://github.com/ZERG3R/image_checker</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>junkdrawer</td>\n",
       "      <td>A digital graveyard for projects that are not of any interest as a whole. Code created for extremely niche uses, canc…</td>\n",
       "      <td>https://github.com/jkmartindale/junkdrawer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>Simple-Bible-Quiz</td>\n",
       "      <td>This is a small app I built for myself to help me study the Bible. Umm, got a really niche crowd.</td>\n",
       "      <td>https://github.com/cmylchan/Simple-Bible-Quiz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>1stpythonproj</td>\n",
       "      <td>This is a small parsing code I decided to make for a list of websites that were being taken down. I wanted to parse t…</td>\n",
       "      <td>https://github.com/BevansMath/1stpythonproj</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>NAlineaGA</td>\n",
       "      <td>An implemention of AineGA With Evolutionary Multiobjective optimization, no parallel yet based on the work of Silva e…</td>\n",
       "      <td>https://github.com/DaveM8/NAlineaGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>MaNGA</td>\n",
       "      <td>MaNGA is a multi-niche/multi-objective genetic algorithm for QSAR modelling, that simultaneously enables stable featu…</td>\n",
       "      <td>https://github.com/angy89/MaNGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>Tweepy-TwitterBot-Beta</td>\n",
       "      <td>My first attempt at building a Twitter Bot in Python which follows back the new followers and tweets a mix of organic…</td>\n",
       "      <td>https://github.com/andreiciulinaru/Tweepy-TwitterBot-Beta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>SEO-tool</td>\n",
       "      <td>This is a python based application that helps analyze a web page to help it's google ranking and it's find-ability gi…</td>\n",
       "      <td>https://github.com/im-vijayts/SEO-tool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>pops</td>\n",
       "      <td>relationships via premier niche markets. Professionally cultivate one-to-one customer service with robust ideas. Dyn…</td>\n",
       "      <td>https://github.com/mitchellzen/pops</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>niche_modelling</td>\n",
       "      <td>Niche modelling with biomod2 using 70 environmental variables summarized in PCA Axes</td>\n",
       "      <td>https://github.com/pedroeisenlohr/niche_modelling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>ENMeval</td>\n",
       "      <td>R package for automated runs and evaluations of ecological niche models.</td>\n",
       "      <td>https://github.com/bobmuscarella/ENMeval</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>ENMGadgets_orig</td>\n",
       "      <td>Gadgets for Ecological Niche Modelling</td>\n",
       "      <td>https://github.com/vijaybarve/ENMGadgets_orig</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>ENM_TheMetaLand</td>\n",
       "      <td>Create Ecological Niche Models with TheMetaLand</td>\n",
       "      <td>https://github.com/andrefaa/ENM_TheMetaLand</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>spaa</td>\n",
       "      <td>Miscellaneous functions for analysis of species association and niche overlap</td>\n",
       "      <td>https://github.com/helixcn/spaa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>Model-R</td>\n",
       "      <td>The shiny application for ModelR, an ecological niche model workflow based on dismo()</td>\n",
       "      <td>https://github.com/Model-R/Model-R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>atlasr</td>\n",
       "      <td>Niche modelling code for the Biogeographic Atlast of the Southern Ocean</td>\n",
       "      <td>https://github.com/jiho/atlasr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>nichenetr</td>\n",
       "      <td>nichenetr</td>\n",
       "      <td>https://github.com/browaeysrobin/nichenetr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>CSC-sky-island-forest</td>\n",
       "      <td>USGS Climate Science Center funded project: fine scale climate downscaling, traits and niche models</td>\n",
       "      <td>https://github.com/schwilklab/CSC-sky-island-forest</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>genome-centric-portrait-of-cellulose-hydrolysis</td>\n",
       "      <td>This pipeline is developed to interpret MAGs on the specific function niche of cellulose hydrolysis</td>\n",
       "      <td>https://github.com/yuboer/genome-centric-portrait-of-cellulose-hydrolysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>marburg_zoonotic</td>\n",
       "      <td>Source code for mapping the zoonotic niche of Marburg virus disease in Africa</td>\n",
       "      <td>https://github.com/SEEG-Oxford/marburg_zoonotic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>anhu</td>\n",
       "      <td>demographic and climate niche analysis of range expansions in the Anna's Hummingbird</td>\n",
       "      <td>https://github.com/cjbattey/anhu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>Mammal-hypervolumes</td>\n",
       "      <td>PhD scripts in R on climatic niche analyses, statistics and figures</td>\n",
       "      <td>https://github.com/Kgiabou/Mammal-hypervolumes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>ACERVNicheSpaceFlorida2011-2015</td>\n",
       "      <td>Project code for \"Niche space of Acropora cervicornis in the Florida reef tract\"</td>\n",
       "      <td>https://github.com/elizabeth201705/ACERVNicheSpaceFlorida2011-2015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>virtualPollen</td>\n",
       "      <td>R package to simulate pollen curves based on virtual taxa with different life and niche traits.</td>\n",
       "      <td>https://github.com/BlasBenito/virtualPollen</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>RSFSA_R</td>\n",
       "      <td>Random subset feature selection algorithm R script for niche modeling with MaxEnt, GLM and GLMnet.</td>\n",
       "      <td>https://github.com/jamesltracy/RSFSA_R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>benmR</td>\n",
       "      <td>Implementation of a Bayesian Method for Estimating Ecological Niches from Presence Data of a Species</td>\n",
       "      <td>https://github.com/LauraJim/benmR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>coastal_ENM</td>\n",
       "      <td>Scripts for \"Ecological niche modeling reveals that climate change promotes mangrove invasion of salt marshes\"</td>\n",
       "      <td>https://github.com/richiehodel/coastal_ENM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>ENM_manuals</td>\n",
       "      <td>R scripts that help to perform common data cleaning process in niche modeling exercises</td>\n",
       "      <td>https://github.com/marlonecobos/ENM_manuals</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>humboldt</td>\n",
       "      <td>Performs spatial analyses of species' niche overlap in e-espace (environment or climate space).</td>\n",
       "      <td>https://github.com/jasonleebrown/humboldt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>nppen</td>\n",
       "      <td>Non-Parametric Probabilistic Ecological Niche from Beaugrand et al 2011 in R.</td>\n",
       "      <td>https://github.com/jiho/nppen</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>Ecological-replacement</td>\n",
       "      <td>A set of analyses to determine niche overlap and potential for natural and deliberate ecological replacement.</td>\n",
       "      <td>https://github.com/SarahEDalrymple/Ecological-replacement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>phyloclim</td>\n",
       "      <td>❗️ This is a read-only mirror of the CRAN R package repository. phyloclim — Integrating Phylogenetics and Climatic Niche</td>\n",
       "      <td>https://github.com/cran/phyloclim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>CENFA</td>\n",
       "      <td>❗️ This is a read-only mirror of the CRAN R package repository. CENFA — Climate and Ecological Niche Factor Analysis</td>\n",
       "      <td>https://github.com/cran/CENFA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>PLNT-model</td>\n",
       "      <td>Phylogenetic Latent Niche Trait (PLNT) Model for modeling (and testing?) phylogenetic structure in environmental filt…</td>\n",
       "      <td>https://github.com/rdinnager/PLNT-model</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>ENM</td>\n",
       "      <td>R scripts to help improve workflow efficiency for ecological niche modelling and provide an introduction for beginners.</td>\n",
       "      <td>https://github.com/LewisAJones/ENM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>envdianthus</td>\n",
       "      <td>Scripts from New Phytologist's manuscript \" Niche divergence and limits to expansion in the high polyploid Dianthus b…</td>\n",
       "      <td>https://github.com/fbalao/envdianthus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>MaxentModelEvaluations</td>\n",
       "      <td>Easy to run scripts for evaluating many combinations of maxent parameters, as well as calculating ecological niche ov…</td>\n",
       "      <td>https://github.com/jscavetta95/MaxentModelEvaluations</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>Strassburg_NEE_ENM</td>\n",
       "      <td>This is the final repository for the ecological niche analyses in Strassburg et al. \"Strategic approaches to restorin…</td>\n",
       "      <td>https://github.com/Model-R/Strassburg_NEE_ENM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>recline</td>\n",
       "      <td>A package written for my PhD thesis research. The one can perform reconstruction of the climatic niche of a species u…</td>\n",
       "      <td>https://github.com/dissipative/recline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>Merow_et_al_2016_GEB_Minxent_Examples</td>\n",
       "      <td>Code to illustrate applications of Minxent models from Merow, Allen, Aiello-Lammens, and Silander. 2016. Improving niche</td>\n",
       "      <td>https://github.com/cmerow/Merow_et_al_2016_GEB_Minxent_Examples</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>Somatic_Evo_DFE</td>\n",
       "      <td>R code associated with The Implications of Small Stem Cell Niche Sizes and the Distribution of Fitness Effects of New…</td>\n",
       "      <td>https://github.com/vcannataro/Somatic_Evo_DFE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>Persistence-Times-Code</td>\n",
       "      <td>R code for \"Biodiversity maintenance may be lower under partial niche differentiation than under neutrality\", by Rafa…</td>\n",
       "      <td>https://github.com/rafaeldandrea/Persistence-Times-Code</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>limlimsim</td>\n",
       "      <td>How does the idea of limiting similarity hold up under niche and fitness differences? Code for Ecology: Theory and Co…</td>\n",
       "      <td>https://github.com/ShanKothari/limlimsim</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>MEGA</td>\n",
       "      <td>MEGA glasshouse experiment with Eucalyptus and Corymbia species. Species were grown in a series of six temperature tr…</td>\n",
       "      <td>https://github.com/jedrake/MEGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>Cariaco-NitrogenCycling</td>\n",
       "      <td>Scripts to go along with manuscript, \"Niche differentiation among free-living and particle-associated nitrogen cyclin…</td>\n",
       "      <td>https://github.com/lizsuter/Cariaco-NitrogenCycling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>MaxentVariableSelection</td>\n",
       "      <td>❗️ This is a read-only mirror of the CRAN R package repository. MaxentVariableSelection — Selecting the Best Set of R…</td>\n",
       "      <td>https://github.com/cran/MaxentVariableSelection</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>NunesPearson2016</td>\n",
       "      <td>Updated R code for Nunes &amp; Pearson, 2016 in Journal of Biogeography. Nunes, L. A. and Pearson, R. G. (2016), A null b…</td>\n",
       "      <td>https://github.com/LauraANunes/NunesPearson2016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>rKIN</td>\n",
       "      <td>❗️ This is a read-only mirror of the CRAN R package repository. rKIN — (Kernel) Isotope Niche Estimation. Homepage: h…</td>\n",
       "      <td>https://github.com/cran/rKIN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>digital-foundry-demand-forcasting</td>\n",
       "      <td>solutions for varying needs of different product types in our product portfolio like new products both evolutionary …</td>\n",
       "      <td>https://github.com/kumarchinnakali/digital-foundry-demand-forcasting</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>nicheTrackR</td>\n",
       "      <td>Do migrant hummingbirds track a climatic niche through the seasons?</td>\n",
       "      <td>https://github.com/cjbattey/nicheTrackR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>ebola_zoonotic</td>\n",
       "      <td>Source code for mapping the zoonotic niche of Ebola virus disease in Africa</td>\n",
       "      <td>https://github.com/SEEG-Oxford/ebola_zoonotic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>biomod2ez</td>\n",
       "      <td>R script suite for use in conjunction with Biomod2 R package for ecological niche ensemble modeling, adding a report …</td>\n",
       "      <td>https://github.com/psesinkclee/biomod2ez</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>WitOMI</td>\n",
       "      <td>MIT license</td>\n",
       "      <td>https://github.com/KarasiewiczStephane/WitOMI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>BridgeTree</td>\n",
       "      <td>For linking and managing occurrence data with phylogenies for niche modeling and ancestral niche reconstructions.</td>\n",
       "      <td>https://github.com/hannahlowens/BridgeTree</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>modelr_pkg</td>\n",
       "      <td>An ecological niche model workflow based on dismo()</td>\n",
       "      <td>https://github.com/Model-R/modelr_pkg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>nicheModeling</td>\n",
       "      <td>Updated Aug 17, 2016</td>\n",
       "      <td>https://github.com/biology-examples/nicheModeling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>NicheRanges</td>\n",
       "      <td>Updated Sep 5, 2018</td>\n",
       "      <td>https://github.com/InesDiogo/NicheRanges</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>nicheROVER</td>\n",
       "      <td>Niche Region and Niche Overlap Metrics for Multidimensional Ecological Niches</td>\n",
       "      <td>https://github.com/mlysy/nicheROVER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>E8-NichePlots</td>\n",
       "      <td>Site conditions are more important than abundance for explaining plant invasion impacts on soil nitrogen cycling</td>\n",
       "      <td>https://github.com/marissalee/E8-NichePlots</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>NicheSimulation</td>\n",
       "      <td>Updated Feb 28, 2016</td>\n",
       "      <td>https://github.com/anwala/NicheSimulation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>NicheModel</td>\n",
       "      <td>R script for building niche models in Maxent for many species.</td>\n",
       "      <td>https://github.com/ccmothes/NicheModel</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>NicheModelling_FBaletaud</td>\n",
       "      <td>Updated Jun 3, 2017</td>\n",
       "      <td>https://github.com/Flbl/NicheModelling_FBaletaud</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>NicheModellingGBIF</td>\n",
       "      <td>Material for a niche model course I teach in GBIF.es every year.</td>\n",
       "      <td>https://github.com/BlasBenito/NicheModellingGBIF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>NicheOverlapR</td>\n",
       "      <td>Ecological niche overlap calculations</td>\n",
       "      <td>https://github.com/JonasGeschke/NicheOverlapR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>Allium_nicheModeling</td>\n",
       "      <td>Updated Oct 8, 2017</td>\n",
       "      <td>https://github.com/aortiz24/Allium_nicheModeling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>nichefillr</td>\n",
       "      <td>R package implementing niche filling simulations</td>\n",
       "      <td>https://github.com/rdinnager/nichefillr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>nichevol</td>\n",
       "      <td>An R package for assessment of evolution of ecological niches</td>\n",
       "      <td>https://github.com/marlonecobos/nichevol</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>nichexplorer</td>\n",
       "      <td>an interactive query and visualization tool for populations of the bone marrow niche</td>\n",
       "      <td>https://github.com/igordot/nichexplorer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>nichetracking</td>\n",
       "      <td>Updated Oct 2, 2018</td>\n",
       "      <td>https://github.com/zeynoon/nichetracking</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>Michela</td>\n",
       "      <td>Horses climatic niche overlap</td>\n",
       "      <td>https://github.com/Kgiabou/Michela</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>galliformes</td>\n",
       "      <td>Galliformes Niche Analysis</td>\n",
       "      <td>https://github.com/shinysolutions/galliformes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>phyloENM</td>\n",
       "      <td>Phylogenetic Environmental Niche Modeling</td>\n",
       "      <td>https://github.com/nmatzke/phyloENM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>insect_niche</td>\n",
       "      <td>Insect Niche shift project</td>\n",
       "      <td>https://github.com/mattecologist/insect_niche</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>niche_hypotheses</td>\n",
       "      <td>Updated Jan 15, 2016</td>\n",
       "      <td>https://github.com/romunov/niche_hypotheses</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>Niche_scripts</td>\n",
       "      <td>R scripts analysing and assessing geograohic variation in dietary niche width of niches</td>\n",
       "      <td>https://github.com/BrianHayden/Niche_scripts</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>DecayNiche</td>\n",
       "      <td>R scripts and analyses for exploring decay niches in Tyson CWD plot.</td>\n",
       "      <td>https://github.com/mshcheg/DecayNiche</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>eco_niche</td>\n",
       "      <td>Updated Jan 25, 2017</td>\n",
       "      <td>https://github.com/erstearns/eco_niche</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>Niche-Modeling-Workshop</td>\n",
       "      <td>Niche-Modeling-Workshop</td>\n",
       "      <td>https://github.com/kevinlove/Niche-Modeling-Workshop</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>niche_models</td>\n",
       "      <td>Updated Jul 27, 2016</td>\n",
       "      <td>https://github.com/loarie/niche_models</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>GerminationNiche</td>\n",
       "      <td>raw data from germination experiments on Mediterranean ruderal forbs 2015-2016 and the R scripts for analysis</td>\n",
       "      <td>https://github.com/Cleonia/GerminationNiche</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>niche_evolution</td>\n",
       "      <td>Updated Jan 27, 2017</td>\n",
       "      <td>https://github.com/jonathanrolland/niche_evolution</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>wost-clim-niche</td>\n",
       "      <td>Joint seasonality in geographic and ecological spaces, illustrated with a partially migratory bird</td>\n",
       "      <td>https://github.com/basille/wost-clim-niche</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>Fire_ant_niche_modeling</td>\n",
       "      <td>Updated Apr 14, 2019</td>\n",
       "      <td>https://github.com/EmmalineSheahan/Fire_ant_niche_modeling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154</th>\n",
       "      <td>Wild-boars</td>\n",
       "      <td>Wild boar ecological niche modeling</td>\n",
       "      <td>https://github.com/lucianolasala/Wild-boars</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155</th>\n",
       "      <td>Chionochloa-niche-evolution</td>\n",
       "      <td>Updated Feb 17, 2019</td>\n",
       "      <td>https://github.com/tufui57/Chionochloa-niche-evolution</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>ENMOD</td>\n",
       "      <td>Ecological Niche Modeling on Docker</td>\n",
       "      <td>https://github.com/ghuertaramos/ENMOD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157</th>\n",
       "      <td>Isotopic-niche-diversity</td>\n",
       "      <td>Spatial analysis of bear-salmon systems in British Columbia, Canada</td>\n",
       "      <td>https://github.com/megansadams/Isotopic-niche-diversity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158</th>\n",
       "      <td>EnvNicheR</td>\n",
       "      <td>❗️ This is a read-only mirror of the CRAN R package repository. EnvNicheR — Niche Estimation</td>\n",
       "      <td>https://github.com/cran/EnvNicheR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>ms_niche_allometric</td>\n",
       "      <td>V3</td>\n",
       "      <td>https://github.com/DominiqueGravel/ms_niche_allometric</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>niche-evolution-project</td>\n",
       "      <td>Updated May 17, 2018</td>\n",
       "      <td>https://github.com/rgerlame/niche-evolution-project</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>mobula_japanica-niche</td>\n",
       "      <td>Niche model for Mobula japanica in the South Eastern Pacific</td>\n",
       "      <td>https://github.com/patricia-alcantara-p/mobula_japanica-niche</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>Ecological_Niche_Modeling</td>\n",
       "      <td>Updated Jul 5, 2018</td>\n",
       "      <td>https://github.com/lindsiebug/Ecological_Niche_Modeling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163</th>\n",
       "      <td>niche.modelR</td>\n",
       "      <td>Updated May 4, 2018</td>\n",
       "      <td>https://github.com/rjBadger/niche.modelR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>SVZ_Aging_Niche</td>\n",
       "      <td>Updated May 30, 2018</td>\n",
       "      <td>https://github.com/bdulken/SVZ_Aging_Niche</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165</th>\n",
       "      <td>Acaena-niche-evolution</td>\n",
       "      <td>Updated May 4, 2019</td>\n",
       "      <td>https://github.com/tufui57/Acaena-niche-evolution</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>Niche-Breadth-Range-Size</td>\n",
       "      <td>This repository contains code and data for analysing niche breadth-range size relationships and simulating climates u…</td>\n",
       "      <td>https://github.com/tem11010/Niche-Breadth-Range-Size</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>sp_dist-niche_modeling</td>\n",
       "      <td>This repository contains the data and scripts that I'm using in my statistical modeling and analysis in my research. …</td>\n",
       "      <td>https://github.com/piyalkarum/sp_dist-niche_modeling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>germcell</td>\n",
       "      <td>Single-cell RNA-seq analysis of human germline cells and their niche cells</td>\n",
       "      <td>https://github.com/zorrodong/germcell</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>Social-Democrats-and-Niche-Parties</td>\n",
       "      <td>For paragrapgh 3.3</td>\n",
       "      <td>https://github.com/everythinginitsrightplace/Social-Democrats-and-Niche-Parties</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>Ecological-niche-modelling</td>\n",
       "      <td>Ecological niche modeling is one of the key research area in the field of ecology and conservation biology. The goal …</td>\n",
       "      <td>https://github.com/shobhana1306/Ecological-niche-modelling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172</th>\n",
       "      <td>asian_niche-1</td>\n",
       "      <td>Repo for d'Alpoim Guedes et al. research on ancient agricultural niches. Trial run for Turkey</td>\n",
       "      <td>https://github.com/jguedes81/asian_niche-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173</th>\n",
       "      <td>RLS-ThermalNiche</td>\n",
       "      <td>Updated Feb 12, 2019</td>\n",
       "      <td>https://github.com/cwaldock1/RLS-ThermalNiche</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>Himalayan-Carex-Climatic-niche-Evolution</td>\n",
       "      <td>Updated Nov 29, 2018</td>\n",
       "      <td>https://github.com/uzma-researcher/Himalayan-Carex-Climatic-niche-Evolution</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175</th>\n",
       "      <td>RLS-thermal-niche-V2</td>\n",
       "      <td>Analysis of realized thermal niche shape and abundance-performance relationships from Reef Life Survey community surv…</td>\n",
       "      <td>https://github.com/cwaldock1/RLS-thermal-niche-V2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>ENFA</td>\n",
       "      <td>Ecological Niche Factor Analysis for my Dissertation</td>\n",
       "      <td>https://github.com/TryAgainTryAgain/ENFA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>ENM_heterogeneity</td>\n",
       "      <td>Working with heterogeneity in Ecological Niche Models</td>\n",
       "      <td>https://github.com/marlonecobos/ENM_heterogeneity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>178</th>\n",
       "      <td>packing</td>\n",
       "      <td>Relative contribution of niche packing and expansion</td>\n",
       "      <td>https://github.com/vpellissier/packing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>kali</td>\n",
       "      <td>Tools for habitat and niche modeling</td>\n",
       "      <td>https://github.com/roliveros-ramos/kali</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>Rossman_etal_2016_EcolAndEvol</td>\n",
       "      <td>Rossman S., Ostrom P.H., Gordon F., and Zipkin E.F. 2016. Beyond carbon and nitrogen: guidelines for estimating three…</td>\n",
       "      <td>https://github.com/zipkinlab/Rossman_etal_2016_EcolAndEvol</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>trochilidae</td>\n",
       "      <td>Global Hummingbird Niche Models and Associated Code</td>\n",
       "      <td>https://github.com/jacobccooper/trochilidae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182</th>\n",
       "      <td>vuln</td>\n",
       "      <td>Niche-based vulnerability of species and communities.</td>\n",
       "      <td>https://github.com/phytomosaic/vuln</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>trait-geo-diverse-ungulates</td>\n",
       "      <td>Comparative analysis of abiotic niches in Ungulates</td>\n",
       "      <td>https://github.com/naturalis/trait-geo-diverse-ungulates</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>Hybrid-niche-modelling-Pacific-Bluefin-tuna--Allgayer</td>\n",
       "      <td>Updated May 1, 2019</td>\n",
       "      <td>https://github.com/rebekkaallgayer/Hybrid-niche-modelling-Pacific-Bluefin-tuna--Allgayer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>Chapter2</td>\n",
       "      <td>All code for chapter 2: Seasonal niche dynamics</td>\n",
       "      <td>https://github.com/alieyres/Chapter2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>Jenior_Modeling_mSystems_2017</td>\n",
       "      <td>Repo associated with Jenior et. al. (2017). Clostridium difficile Colonizes Alternative Nutrient Niches during Infect…</td>\n",
       "      <td>https://github.com/SchlossLab/Jenior_Modeling_mSystems_2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>giantPetrels</td>\n",
       "      <td>Habitat and trophic niche segregation in giant petrels</td>\n",
       "      <td>https://github.com/ryanreisinger/giantPetrels</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>Bucanetes</td>\n",
       "      <td>R-script of the ecological niche model for Bucanetes githagineus</td>\n",
       "      <td>https://github.com/SaraVarela/Bucanetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>futurescenarios</td>\n",
       "      <td>Niche modelling with biomod2 (Future scenarios of climate change)</td>\n",
       "      <td>https://github.com/pedroeisenlohr/futurescenarios</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190</th>\n",
       "      <td>FranceCG</td>\n",
       "      <td>Field common garden and climate niche analysis of Centaurea diffusa.</td>\n",
       "      <td>https://github.com/kgturner/FranceCG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>lump.split.pool.ENM</td>\n",
       "      <td>Running simulations to compare different approaches for ecological niche modelling</td>\n",
       "      <td>https://github.com/Pakillo/lump.split.pool.ENM</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                      name  \\\n",
       "26   CENFA                                                   \n",
       "27   ENiRG                                                   \n",
       "28   ENMeval                                                 \n",
       "29   EnvNicheR                                               \n",
       "30   MaxentVariableSelection                                 \n",
       "31   nicheROVER                                              \n",
       "32   phyloclim                                               \n",
       "33   rKIN                                                    \n",
       "0    xpost-bot                                               \n",
       "2    fynat                                                   \n",
       "3    ambitus                                                 \n",
       "4    opalescence                                             \n",
       "5    niche_vlaanderen                                        \n",
       "6    NicheLife                                               \n",
       "7    nicheLang                                               \n",
       "8    language-niche-research                                 \n",
       "9    NichePSO                                                \n",
       "10   niche                                                   \n",
       "11   nicheCrawler                                            \n",
       "12   NICHE                                                   \n",
       "13   NicheReviewScraperPY                                    \n",
       "14   nicheRecommender                                        \n",
       "15   Nichehire                                               \n",
       "16   nicheapps                                               \n",
       "17   niches_finder                                           \n",
       "18   Coding-niche                                            \n",
       "19   MyNiche                                                 \n",
       "20   niche_modeling                                          \n",
       "22   nndocker                                                \n",
       "23   app-niche                                               \n",
       "24   niche-scraping                                          \n",
       "25   niche-construction                                      \n",
       "26   Serendipity                                             \n",
       "27   api.niche.fm                                            \n",
       "28   powers_niche_construction                               \n",
       "29   Temporal_niche_modeling                                 \n",
       "30   Random-Niche-Generation                                 \n",
       "31   lifeNicheTemp                                           \n",
       "32   Niche-EnterpriseProject                                 \n",
       "33   Niche-School-Scraper                                    \n",
       "34   Instagram-Niche-Downloader                              \n",
       "35   netyell                                                 \n",
       "36   Python-Scripts                                          \n",
       "37   DNAP                                                    \n",
       "38   GA_Lab                                                  \n",
       "39   hong_kong_waiting_time_reused_niches                    \n",
       "40   Differential-Evolution                                  \n",
       "41   peutils                                                 \n",
       "42   PCSDswag                                                \n",
       "43   NMMSO                                                   \n",
       "44   EntheoBot                                               \n",
       "45   EasyList                                                \n",
       "46   instaBot                                                \n",
       "47   moneyball_app                                           \n",
       "48   opensim-stl-export                                      \n",
       "49   CTCI                                                    \n",
       "50   US-Universities-Scraper                                 \n",
       "51   blender-project-tools                                   \n",
       "52   MunichBFOR                                              \n",
       "53   MusicAnalysisTools                                      \n",
       "54   write_for_us                                            \n",
       "55   multi-modal-ga                                          \n",
       "56   fuzzy_habitat_modelling                                 \n",
       "57   Youtuber-Discovery                                      \n",
       "58   MAGA                                                    \n",
       "59   SelfDestructBot                                         \n",
       "60   BATty                                                   \n",
       "61   WallpaperSort                                           \n",
       "62   CommNetAntennasExtension                                \n",
       "63   EccentricRecommender                                    \n",
       "64   image_checker                                           \n",
       "65   junkdrawer                                              \n",
       "66   Simple-Bible-Quiz                                       \n",
       "67   1stpythonproj                                           \n",
       "68   NAlineaGA                                               \n",
       "69   MaNGA                                                   \n",
       "70   Tweepy-TwitterBot-Beta                                  \n",
       "71   SEO-tool                                                \n",
       "72   pops                                                    \n",
       "74   niche_modelling                                         \n",
       "75   ENMeval                                                 \n",
       "76   ENMGadgets_orig                                         \n",
       "77   ENM_TheMetaLand                                         \n",
       "79   spaa                                                    \n",
       "80   Model-R                                                 \n",
       "81   atlasr                                                  \n",
       "82   nichenetr                                               \n",
       "83   CSC-sky-island-forest                                   \n",
       "84   genome-centric-portrait-of-cellulose-hydrolysis         \n",
       "86   marburg_zoonotic                                        \n",
       "87   anhu                                                    \n",
       "88   Mammal-hypervolumes                                     \n",
       "89   ACERVNicheSpaceFlorida2011-2015                         \n",
       "90   virtualPollen                                           \n",
       "91   RSFSA_R                                                 \n",
       "92   benmR                                                   \n",
       "93   coastal_ENM                                             \n",
       "94   ENM_manuals                                             \n",
       "95   humboldt                                                \n",
       "96   nppen                                                   \n",
       "97   Ecological-replacement                                  \n",
       "98   phyloclim                                               \n",
       "99   CENFA                                                   \n",
       "100  PLNT-model                                              \n",
       "101  ENM                                                     \n",
       "102  envdianthus                                             \n",
       "103  MaxentModelEvaluations                                  \n",
       "105  Strassburg_NEE_ENM                                      \n",
       "106  recline                                                 \n",
       "107  Merow_et_al_2016_GEB_Minxent_Examples                   \n",
       "108  Somatic_Evo_DFE                                         \n",
       "109  Persistence-Times-Code                                  \n",
       "110  limlimsim                                               \n",
       "111  MEGA                                                    \n",
       "112  Cariaco-NitrogenCycling                                 \n",
       "113  MaxentVariableSelection                                 \n",
       "114  NunesPearson2016                                        \n",
       "115  rKIN                                                    \n",
       "116  digital-foundry-demand-forcasting                       \n",
       "118  nicheTrackR                                             \n",
       "119  ebola_zoonotic                                          \n",
       "120  biomod2ez                                               \n",
       "122  WitOMI                                                  \n",
       "123  BridgeTree                                              \n",
       "124  modelr_pkg                                              \n",
       "126  nicheModeling                                           \n",
       "127  NicheRanges                                             \n",
       "128  nicheROVER                                              \n",
       "129  E8-NichePlots                                           \n",
       "130  NicheSimulation                                         \n",
       "131  NicheModel                                              \n",
       "132  NicheModelling_FBaletaud                                \n",
       "133  NicheModellingGBIF                                      \n",
       "134  NicheOverlapR                                           \n",
       "135  Allium_nicheModeling                                    \n",
       "136  nichefillr                                              \n",
       "137  nichevol                                                \n",
       "138  nichexplorer                                            \n",
       "139  nichetracking                                           \n",
       "140  Michela                                                 \n",
       "141  galliformes                                             \n",
       "142  phyloENM                                                \n",
       "143  insect_niche                                            \n",
       "144  niche_hypotheses                                        \n",
       "145  Niche_scripts                                           \n",
       "146  DecayNiche                                              \n",
       "147  eco_niche                                               \n",
       "148  Niche-Modeling-Workshop                                 \n",
       "149  niche_models                                            \n",
       "150  GerminationNiche                                        \n",
       "151  niche_evolution                                         \n",
       "152  wost-clim-niche                                         \n",
       "153  Fire_ant_niche_modeling                                 \n",
       "154  Wild-boars                                              \n",
       "155  Chionochloa-niche-evolution                             \n",
       "156  ENMOD                                                   \n",
       "157  Isotopic-niche-diversity                                \n",
       "158  EnvNicheR                                               \n",
       "159  ms_niche_allometric                                     \n",
       "160  niche-evolution-project                                 \n",
       "161  mobula_japanica-niche                                   \n",
       "162  Ecological_Niche_Modeling                               \n",
       "163  niche.modelR                                            \n",
       "164  SVZ_Aging_Niche                                         \n",
       "165  Acaena-niche-evolution                                  \n",
       "166  Niche-Breadth-Range-Size                                \n",
       "168  sp_dist-niche_modeling                                  \n",
       "169  germcell                                                \n",
       "170  Social-Democrats-and-Niche-Parties                      \n",
       "171  Ecological-niche-modelling                              \n",
       "172  asian_niche-1                                           \n",
       "173  RLS-ThermalNiche                                        \n",
       "174  Himalayan-Carex-Climatic-niche-Evolution                \n",
       "175  RLS-thermal-niche-V2                                    \n",
       "176  ENFA                                                    \n",
       "177  ENM_heterogeneity                                       \n",
       "178  packing                                                 \n",
       "179  kali                                                    \n",
       "180  Rossman_etal_2016_EcolAndEvol                           \n",
       "181  trochilidae                                             \n",
       "182  vuln                                                    \n",
       "183  trait-geo-diverse-ungulates                             \n",
       "184  Hybrid-niche-modelling-Pacific-Bluefin-tuna--Allgayer   \n",
       "185  Chapter2                                                \n",
       "186  Jenior_Modeling_mSystems_2017                           \n",
       "187  giantPetrels                                            \n",
       "188  Bucanetes                                               \n",
       "189  futurescenarios                                         \n",
       "190  FranceCG                                                \n",
       "191  lump.split.pool.ENM                                     \n",
       "\n",
       "                                                                                                                               description  \\\n",
       "26   Climate and Ecological Niche Factor Analysis                                                                                            \n",
       "27   Ecological Niche in R and GRASS                                                                                                         \n",
       "28   Automated Runs and Evaluations of Ecological Niche Models                                                                               \n",
       "29   Niche Estimation                                                                                                                        \n",
       "30   Selecting the Best Set of Relevant Environmental Variables along with the Optimal Regularization Multiplier for Maxent Niche Modeling   \n",
       "31   (Niche) (R)egion and Niche (Over)lap Metrics for Multidimensional Ecological Niches                                                     \n",
       "32   Integrating Phylogenetics and Climatic Niche Modeling                                                                                   \n",
       "33   (Kernel) Isotope Niche Estimation                                                                                                       \n",
       "0    Reddit bot to scan and repost submissions of interest to niche subreddits                                                               \n",
       "2    Find Your Niche At Tech                                                                                                                 \n",
       "3    Ancestral niche reconstruction and associated tools.                                                                                    \n",
       "4    Niche-purpose discord moderation utility bot                                                                                            \n",
       "5    Python package to run the NICHE Vlaanderen model                                                                                        \n",
       "6    A map visualization tool to find your ideal neighborhoods in Manhattan                                                                  \n",
       "7    Updated May 22, 2018                                                                                                                    \n",
       "8    Scientific repository for fiddling around with linguistic data                                                                          \n",
       "9    Updated Aug 22, 2018                                                                                                                    \n",
       "10   MIT license                                                                                                                             \n",
       "11   nicheCrawler is a bot that crawls through niche.com and returns all universities from desired state.                                    \n",
       "12   Code for UNICEF NICHE project.                                                                                                          \n",
       "13   Updated Oct 22, 2015                                                                                                                    \n",
       "14   Provides niche, lesser-known places for people to visit, explore, eat at, etc.                                                          \n",
       "15   Updated Dec 6, 2017                                                                                                                     \n",
       "16   nicheapps is a platform to showcase your apps to the users , post a demo , get feedback                                                 \n",
       "17   Updated Mar 21, 2019                                                                                                                    \n",
       "18   Updated Oct 14, 2018                                                                                                                    \n",
       "19   Website built to allow clubs to share notifications in real time with school students. Manages club accounts and stud…                  \n",
       "20   Updated Sep 3, 2014                                                                                                                     \n",
       "22   Neural Niche Docker Config                                                                                                              \n",
       "23   API RESTful to find your app niche in Flask                                                                                             \n",
       "24   Updated Mar 30, 2019                                                                                                                    \n",
       "25   Studies on niche construction through agent based modeling.                                                                             \n",
       "26   /r/serendipity is a meta-subreddit meant to broaden the perspective of its subscribers. It takes a popular entry from…                  \n",
       "27   Updated Mar 17, 2018                                                                                                                    \n",
       "28   Powers et al's niche construction model                                                                                                 \n",
       "29   Updated Feb 14, 2018                                                                                                                    \n",
       "30   Randomly creates a sequence of niches within the full nuclear fuel cycle based on a dictionary of plausible connections.                \n",
       "31   lifeNicheTemp                                                                                                                           \n",
       "32   Updated Apr 19, 2017                                                                                                                    \n",
       "33   A Python script that uses BeautifulSoup and requests to scrape the addresses from high schools on niche.com.                            \n",
       "34   Download images content as per your niche.                                                                                              \n",
       "35   sends formatted messages to symon netbrite signs from the command line. (yep, there's a niche!)                                         \n",
       "36   Small python scripts for niche uses.                                                                                                    \n",
       "37   Release checker for niche vinyl labels                                                                                                  \n",
       "38   Genetic algorithm niching functions lab work                                                                                            \n",
       "39   Hong Kong Waiting Time for Allocation of Reused Niches                                                                                  \n",
       "40   差分进化算法，支持小生境（niche）技术，支持多目标                                                                                                             \n",
       "41   Binary Ninja plugin providing various niche utilities for working with PE binaries                                                      \n",
       "42   A niche screen scraper for just one (1) website                                                                                         \n",
       "43   Python Implementation of the Niching Migratory Multi-Swarm Optimizer                                                                    \n",
       "44   Discord bot for a niche group of chatters.                                                                                              \n",
       "45   A niche type of list keeper in python.                                                                                                  \n",
       "46   upgraded tool to follow an like users posts on your niche . u need to enter your username and password then your niche …                \n",
       "47   A webapp to track results of foosball games in financial institutions. It is a very niche market!                                       \n",
       "48   Want to export a model from OpenSim to STL? Wegotchufam.                                                                                \n",
       "49   interview practice from CTCI, leetcode, hackerrank, and other niches of the web                                                         \n",
       "50   Scraper that provides information on universities from various sites(ex. USNews, Niche)                                                 \n",
       "51   Code & tools made to assist with various niche Moo-Ack! Productions projects                                                            \n",
       "52   Evolutionary Multi Niche Bacterial Foraging Algorithm for Multimodal Dynamic Optimizations and Neural Network Optimiz…                  \n",
       "53   music21 based tools for analysis. Often WIP and / or too niche for the main music21 repo                                                \n",
       "54   Script for finding niche websites via Google search that accept article contributions                                                   \n",
       "55   A multi-modal genetic algorithm which will use two niching methods.                                                                     \n",
       "56   Fish habitat model and visualization of habitat niche using GRASS GIS r.fuzzy.system and R                                              \n",
       "57   Single page web app for finding or researching youtubers in a given niche or area                                                       \n",
       "58   A Multi-niche Adaptive Genetic Algorithm for biomarker discovery from omics data.                                                       \n",
       "59   A niche moderator bot that removes posts after a specified amount of time.                                                              \n",
       "60   A Python powered serverless web app for understanding the current market value of niche used cars                                       \n",
       "61   A Python script to sort 1.6 and 1.7 wallpapers into separate folders. Kinda niche, to be honest.                                        \n",
       "62   Add 6 new antennas for CommNet, filling a niche in the transmission ranges: direct 5M, relay 32M, relays 7G and 25G, …                  \n",
       "63   Using a new Metric called Eccentricity this Recommender System aims on improving the variations in modern Recommendat…                  \n",
       "64   Used for checking one locations files against another locations files to find the difference (if one exists), extremely                 \n",
       "65   A digital graveyard for projects that are not of any interest as a whole. Code created for extremely niche uses, canc…                  \n",
       "66   This is a small app I built for myself to help me study the Bible. Umm, got a really niche crowd.                                       \n",
       "67   This is a small parsing code I decided to make for a list of websites that were being taken down. I wanted to parse t…                  \n",
       "68   An implemention of AineGA With Evolutionary Multiobjective optimization, no parallel yet based on the work of Silva e…                  \n",
       "69   MaNGA is a multi-niche/multi-objective genetic algorithm for QSAR modelling, that simultaneously enables stable featu…                  \n",
       "70   My first attempt at building a Twitter Bot in Python which follows back the new followers and tweets a mix of organic…                  \n",
       "71   This is a python based application that helps analyze a web page to help it's google ranking and it's find-ability gi…                  \n",
       "72   relationships via premier niche markets. Professionally cultivate one-to-one customer service with robust ideas. Dyn…                   \n",
       "74   Niche modelling with biomod2 using 70 environmental variables summarized in PCA Axes                                                    \n",
       "75   R package for automated runs and evaluations of ecological niche models.                                                                \n",
       "76   Gadgets for Ecological Niche Modelling                                                                                                  \n",
       "77   Create Ecological Niche Models with TheMetaLand                                                                                         \n",
       "79   Miscellaneous functions for analysis of species association and niche overlap                                                           \n",
       "80   The shiny application for ModelR, an ecological niche model workflow based on dismo()                                                   \n",
       "81   Niche modelling code for the Biogeographic Atlast of the Southern Ocean                                                                 \n",
       "82   nichenetr                                                                                                                               \n",
       "83   USGS Climate Science Center funded project: fine scale climate downscaling, traits and niche models                                     \n",
       "84   This pipeline is developed to interpret MAGs on the specific function niche of cellulose hydrolysis                                     \n",
       "86   Source code for mapping the zoonotic niche of Marburg virus disease in Africa                                                           \n",
       "87   demographic and climate niche analysis of range expansions in the Anna's Hummingbird                                                    \n",
       "88   PhD scripts in R on climatic niche analyses, statistics and figures                                                                     \n",
       "89   Project code for \"Niche space of Acropora cervicornis in the Florida reef tract\"                                                        \n",
       "90   R package to simulate pollen curves based on virtual taxa with different life and niche traits.                                         \n",
       "91   Random subset feature selection algorithm R script for niche modeling with MaxEnt, GLM and GLMnet.                                      \n",
       "92   Implementation of a Bayesian Method for Estimating Ecological Niches from Presence Data of a Species                                    \n",
       "93   Scripts for \"Ecological niche modeling reveals that climate change promotes mangrove invasion of salt marshes\"                          \n",
       "94   R scripts that help to perform common data cleaning process in niche modeling exercises                                                 \n",
       "95   Performs spatial analyses of species' niche overlap in e-espace (environment or climate space).                                         \n",
       "96   Non-Parametric Probabilistic Ecological Niche from Beaugrand et al 2011 in R.                                                           \n",
       "97   A set of analyses to determine niche overlap and potential for natural and deliberate ecological replacement.                           \n",
       "98   ❗️ This is a read-only mirror of the CRAN R package repository. phyloclim — Integrating Phylogenetics and Climatic Niche                \n",
       "99   ❗️ This is a read-only mirror of the CRAN R package repository. CENFA — Climate and Ecological Niche Factor Analysis                    \n",
       "100  Phylogenetic Latent Niche Trait (PLNT) Model for modeling (and testing?) phylogenetic structure in environmental filt…                  \n",
       "101  R scripts to help improve workflow efficiency for ecological niche modelling and provide an introduction for beginners.                 \n",
       "102  Scripts from New Phytologist's manuscript \" Niche divergence and limits to expansion in the high polyploid Dianthus b…                  \n",
       "103  Easy to run scripts for evaluating many combinations of maxent parameters, as well as calculating ecological niche ov…                  \n",
       "105  This is the final repository for the ecological niche analyses in Strassburg et al. \"Strategic approaches to restorin…                  \n",
       "106  A package written for my PhD thesis research. The one can perform reconstruction of the climatic niche of a species u…                  \n",
       "107  Code to illustrate applications of Minxent models from Merow, Allen, Aiello-Lammens, and Silander. 2016. Improving niche                \n",
       "108  R code associated with The Implications of Small Stem Cell Niche Sizes and the Distribution of Fitness Effects of New…                  \n",
       "109  R code for \"Biodiversity maintenance may be lower under partial niche differentiation than under neutrality\", by Rafa…                  \n",
       "110  How does the idea of limiting similarity hold up under niche and fitness differences? Code for Ecology: Theory and Co…                  \n",
       "111  MEGA glasshouse experiment with Eucalyptus and Corymbia species. Species were grown in a series of six temperature tr…                  \n",
       "112  Scripts to go along with manuscript, \"Niche differentiation among free-living and particle-associated nitrogen cyclin…                  \n",
       "113  ❗️ This is a read-only mirror of the CRAN R package repository. MaxentVariableSelection — Selecting the Best Set of R…                  \n",
       "114  Updated R code for Nunes & Pearson, 2016 in Journal of Biogeography. Nunes, L. A. and Pearson, R. G. (2016), A null b…                  \n",
       "115  ❗️ This is a read-only mirror of the CRAN R package repository. rKIN — (Kernel) Isotope Niche Estimation. Homepage: h…                  \n",
       "116  solutions for varying needs of different product types in our product portfolio like new products both evolutionary …                   \n",
       "118  Do migrant hummingbirds track a climatic niche through the seasons?                                                                     \n",
       "119  Source code for mapping the zoonotic niche of Ebola virus disease in Africa                                                             \n",
       "120  R script suite for use in conjunction with Biomod2 R package for ecological niche ensemble modeling, adding a report …                  \n",
       "122  MIT license                                                                                                                             \n",
       "123  For linking and managing occurrence data with phylogenies for niche modeling and ancestral niche reconstructions.                       \n",
       "124  An ecological niche model workflow based on dismo()                                                                                     \n",
       "126  Updated Aug 17, 2016                                                                                                                    \n",
       "127  Updated Sep 5, 2018                                                                                                                     \n",
       "128  Niche Region and Niche Overlap Metrics for Multidimensional Ecological Niches                                                           \n",
       "129  Site conditions are more important than abundance for explaining plant invasion impacts on soil nitrogen cycling                        \n",
       "130  Updated Feb 28, 2016                                                                                                                    \n",
       "131  R script for building niche models in Maxent for many species.                                                                          \n",
       "132  Updated Jun 3, 2017                                                                                                                     \n",
       "133  Material for a niche model course I teach in GBIF.es every year.                                                                        \n",
       "134  Ecological niche overlap calculations                                                                                                   \n",
       "135  Updated Oct 8, 2017                                                                                                                     \n",
       "136  R package implementing niche filling simulations                                                                                        \n",
       "137  An R package for assessment of evolution of ecological niches                                                                           \n",
       "138  an interactive query and visualization tool for populations of the bone marrow niche                                                    \n",
       "139  Updated Oct 2, 2018                                                                                                                     \n",
       "140  Horses climatic niche overlap                                                                                                           \n",
       "141  Galliformes Niche Analysis                                                                                                              \n",
       "142  Phylogenetic Environmental Niche Modeling                                                                                               \n",
       "143  Insect Niche shift project                                                                                                              \n",
       "144  Updated Jan 15, 2016                                                                                                                    \n",
       "145  R scripts analysing and assessing geograohic variation in dietary niche width of niches                                                 \n",
       "146  R scripts and analyses for exploring decay niches in Tyson CWD plot.                                                                    \n",
       "147  Updated Jan 25, 2017                                                                                                                    \n",
       "148  Niche-Modeling-Workshop                                                                                                                 \n",
       "149  Updated Jul 27, 2016                                                                                                                    \n",
       "150  raw data from germination experiments on Mediterranean ruderal forbs 2015-2016 and the R scripts for analysis                           \n",
       "151  Updated Jan 27, 2017                                                                                                                    \n",
       "152  Joint seasonality in geographic and ecological spaces, illustrated with a partially migratory bird                                      \n",
       "153  Updated Apr 14, 2019                                                                                                                    \n",
       "154  Wild boar ecological niche modeling                                                                                                     \n",
       "155  Updated Feb 17, 2019                                                                                                                    \n",
       "156  Ecological Niche Modeling on Docker                                                                                                     \n",
       "157  Spatial analysis of bear-salmon systems in British Columbia, Canada                                                                     \n",
       "158  ❗️ This is a read-only mirror of the CRAN R package repository. EnvNicheR — Niche Estimation                                            \n",
       "159  V3                                                                                                                                      \n",
       "160  Updated May 17, 2018                                                                                                                    \n",
       "161  Niche model for Mobula japanica in the South Eastern Pacific                                                                            \n",
       "162  Updated Jul 5, 2018                                                                                                                     \n",
       "163  Updated May 4, 2018                                                                                                                     \n",
       "164  Updated May 30, 2018                                                                                                                    \n",
       "165  Updated May 4, 2019                                                                                                                     \n",
       "166  This repository contains code and data for analysing niche breadth-range size relationships and simulating climates u…                  \n",
       "168  This repository contains the data and scripts that I'm using in my statistical modeling and analysis in my research. …                  \n",
       "169  Single-cell RNA-seq analysis of human germline cells and their niche cells                                                              \n",
       "170  For paragrapgh 3.3                                                                                                                      \n",
       "171  Ecological niche modeling is one of the key research area in the field of ecology and conservation biology. The goal …                  \n",
       "172  Repo for d'Alpoim Guedes et al. research on ancient agricultural niches. Trial run for Turkey                                           \n",
       "173  Updated Feb 12, 2019                                                                                                                    \n",
       "174  Updated Nov 29, 2018                                                                                                                    \n",
       "175  Analysis of realized thermal niche shape and abundance-performance relationships from Reef Life Survey community surv…                  \n",
       "176  Ecological Niche Factor Analysis for my Dissertation                                                                                    \n",
       "177  Working with heterogeneity in Ecological Niche Models                                                                                   \n",
       "178  Relative contribution of niche packing and expansion                                                                                    \n",
       "179  Tools for habitat and niche modeling                                                                                                    \n",
       "180  Rossman S., Ostrom P.H., Gordon F., and Zipkin E.F. 2016. Beyond carbon and nitrogen: guidelines for estimating three…                  \n",
       "181  Global Hummingbird Niche Models and Associated Code                                                                                     \n",
       "182  Niche-based vulnerability of species and communities.                                                                                   \n",
       "183  Comparative analysis of abiotic niches in Ungulates                                                                                     \n",
       "184  Updated May 1, 2019                                                                                                                     \n",
       "185  All code for chapter 2: Seasonal niche dynamics                                                                                         \n",
       "186  Repo associated with Jenior et. al. (2017). Clostridium difficile Colonizes Alternative Nutrient Niches during Infect…                  \n",
       "187  Habitat and trophic niche segregation in giant petrels                                                                                  \n",
       "188  R-script of the ecological niche model for Bucanetes githagineus                                                                        \n",
       "189  Niche modelling with biomod2 (Future scenarios of climate change)                                                                       \n",
       "190  Field common garden and climate niche analysis of Centaurea diffusa.                                                                    \n",
       "191  Running simulations to compare different approaches for ecological niche modelling                                                      \n",
       "\n",
       "                                                                                          url  \n",
       "26   https://cran.r-project.org/web/packages/CENFA/index.html                                  \n",
       "27   https://cran.r-project.org/web/packages/ENiRG/index.html                                  \n",
       "28   https://cran.r-project.org/web/packages/ENMeval/index.html                                \n",
       "29   https://cran.r-project.org/web/packages/EnvNicheR/index.html                              \n",
       "30   https://cran.r-project.org/web/packages/MaxentVariableSelection/index.html                \n",
       "31   https://cran.r-project.org/web/packages/nicheROVER/index.html                             \n",
       "32   https://cran.r-project.org/web/packages/phyloclim/index.html                              \n",
       "33   https://cran.r-project.org/web/packages/rKIN/index.html                                   \n",
       "0    https://github.com/git2samus/xpost-bot                                                    \n",
       "2    https://github.com/dadams22/fynat                                                         \n",
       "3    https://github.com/ryanafolk/ambitus                                                      \n",
       "4    https://github.com/NiteAx/opalescence                                                     \n",
       "5    https://github.com/inbo/niche_vlaanderen                                                  \n",
       "6    https://github.com/priya-vijay/NicheLife                                                  \n",
       "7    https://github.com/mattHoyDev/nicheLang                                                   \n",
       "8    https://github.com/RichardLitt/language-niche-research                                    \n",
       "9    https://github.com/bryaneduardo24/NichePSO                                                \n",
       "10   https://github.com/rwsharp/niche                                                          \n",
       "11   https://github.com/onthir/nicheCrawler                                                    \n",
       "12   https://github.com/alison-campion/NICHE                                                   \n",
       "13   https://github.com/SazanDauti/NicheReviewScraperPY                                        \n",
       "14   https://github.com/Qwertazertl0/nicheRecommender                                          \n",
       "15   https://github.com/saikeerthana19/Nichehire                                               \n",
       "16   https://github.com/addyrookie/nicheapps                                                   \n",
       "17   https://github.com/lukadamczyk/niches_finder                                              \n",
       "18   https://github.com/JayatiGoyal/Coding-niche                                               \n",
       "19   https://github.com/aiswaryasankar/MyNiche                                                 \n",
       "20   https://github.com/tomquisel/niche_modeling                                               \n",
       "22   https://github.com/grahamannett/nndocker                                                  \n",
       "23   https://github.com/costular/app-niche                                                     \n",
       "24   https://github.com/rydena021/niche-scraping                                               \n",
       "25   https://github.com/CA-Lab/niche-construction                                              \n",
       "26   https://github.com/umbrae/Serendipity                                                     \n",
       "27   https://github.com/hery/api.niche.fm                                                      \n",
       "28   https://github.com/greenape/powers_niche_construction                                     \n",
       "29   https://github.com/gaodifan/Temporal_niche_modeling                                       \n",
       "30   https://github.com/aglenn2/Random-Niche-Generation                                        \n",
       "31   https://github.com/Bouteloua/lifeNicheTemp                                                \n",
       "32   https://github.com/vinaykl/Niche-EnterpriseProject                                        \n",
       "33   https://github.com/haanmiba/Niche-School-Scraper                                          \n",
       "34   https://github.com/ougrocks/Instagram-Niche-Downloader                                    \n",
       "35   https://github.com/rsaxvc/netyell                                                         \n",
       "36   https://github.com/keshavd/Python-Scripts                                                 \n",
       "37   https://github.com/Tenchi2xh/DNAP                                                         \n",
       "38   https://github.com/watemerald/GA_Lab                                                      \n",
       "39   https://github.com/howawong/hong_kong_waiting_time_reused_niches                          \n",
       "40   https://github.com/sgdd66/Differential-Evolution                                          \n",
       "41   https://github.com/404d/peutils                                                           \n",
       "42   https://github.com/macsj200/PCSDswag                                                      \n",
       "43   https://github.com/varchasgopalaswamy/NMMSO                                               \n",
       "44   https://github.com/DrewTChrist/EntheoBot                                                  \n",
       "45   https://github.com/wesselmeijer/EasyList                                                  \n",
       "46   https://github.com/crhexn8belda/instaBot                                                  \n",
       "47   https://github.com/moneyball-dk/moneyball_app                                             \n",
       "48   https://github.com/prnthp/opensim-stl-export                                              \n",
       "49   https://github.com/jazzminewang/CTCI                                                      \n",
       "50   https://github.com/Jenniferrrrrr/US-Universities-Scraper                                  \n",
       "51   https://github.com/TheDuckCow/blender-project-tools                                       \n",
       "52   https://github.com/dananjayamahesh/MunichBFOR                                             \n",
       "53   https://github.com/MarkGotham/MusicAnalysisTools                                          \n",
       "54   https://github.com/davejonesbkk/write_for_us                                              \n",
       "55   https://github.com/serbanc94/multi-modal-ga                                               \n",
       "56   https://github.com/jradinger/fuzzy_habitat_modelling                                      \n",
       "57   https://github.com/afimbufe/Youtuber-Discovery                                            \n",
       "58   https://github.com/Greco-Lab/MAGA                                                         \n",
       "59   https://github.com/diceroll123/SelfDestructBot                                            \n",
       "60   https://github.com/dhicks43/BATty                                                         \n",
       "61   https://github.com/KBKarma/WallpaperSort                                                  \n",
       "62   https://github.com/yalov/CommNetAntennasExtension                                         \n",
       "63   https://github.com/SandroLuck/EccentricRecommender                                        \n",
       "64   https://github.com/ZERG3R/image_checker                                                   \n",
       "65   https://github.com/jkmartindale/junkdrawer                                                \n",
       "66   https://github.com/cmylchan/Simple-Bible-Quiz                                             \n",
       "67   https://github.com/BevansMath/1stpythonproj                                               \n",
       "68   https://github.com/DaveM8/NAlineaGA                                                       \n",
       "69   https://github.com/angy89/MaNGA                                                           \n",
       "70   https://github.com/andreiciulinaru/Tweepy-TwitterBot-Beta                                 \n",
       "71   https://github.com/im-vijayts/SEO-tool                                                    \n",
       "72   https://github.com/mitchellzen/pops                                                       \n",
       "74   https://github.com/pedroeisenlohr/niche_modelling                                         \n",
       "75   https://github.com/bobmuscarella/ENMeval                                                  \n",
       "76   https://github.com/vijaybarve/ENMGadgets_orig                                             \n",
       "77   https://github.com/andrefaa/ENM_TheMetaLand                                               \n",
       "79   https://github.com/helixcn/spaa                                                           \n",
       "80   https://github.com/Model-R/Model-R                                                        \n",
       "81   https://github.com/jiho/atlasr                                                            \n",
       "82   https://github.com/browaeysrobin/nichenetr                                                \n",
       "83   https://github.com/schwilklab/CSC-sky-island-forest                                       \n",
       "84   https://github.com/yuboer/genome-centric-portrait-of-cellulose-hydrolysis                 \n",
       "86   https://github.com/SEEG-Oxford/marburg_zoonotic                                           \n",
       "87   https://github.com/cjbattey/anhu                                                          \n",
       "88   https://github.com/Kgiabou/Mammal-hypervolumes                                            \n",
       "89   https://github.com/elizabeth201705/ACERVNicheSpaceFlorida2011-2015                        \n",
       "90   https://github.com/BlasBenito/virtualPollen                                               \n",
       "91   https://github.com/jamesltracy/RSFSA_R                                                    \n",
       "92   https://github.com/LauraJim/benmR                                                         \n",
       "93   https://github.com/richiehodel/coastal_ENM                                                \n",
       "94   https://github.com/marlonecobos/ENM_manuals                                               \n",
       "95   https://github.com/jasonleebrown/humboldt                                                 \n",
       "96   https://github.com/jiho/nppen                                                             \n",
       "97   https://github.com/SarahEDalrymple/Ecological-replacement                                 \n",
       "98   https://github.com/cran/phyloclim                                                         \n",
       "99   https://github.com/cran/CENFA                                                             \n",
       "100  https://github.com/rdinnager/PLNT-model                                                   \n",
       "101  https://github.com/LewisAJones/ENM                                                        \n",
       "102  https://github.com/fbalao/envdianthus                                                     \n",
       "103  https://github.com/jscavetta95/MaxentModelEvaluations                                     \n",
       "105  https://github.com/Model-R/Strassburg_NEE_ENM                                             \n",
       "106  https://github.com/dissipative/recline                                                    \n",
       "107  https://github.com/cmerow/Merow_et_al_2016_GEB_Minxent_Examples                           \n",
       "108  https://github.com/vcannataro/Somatic_Evo_DFE                                             \n",
       "109  https://github.com/rafaeldandrea/Persistence-Times-Code                                   \n",
       "110  https://github.com/ShanKothari/limlimsim                                                  \n",
       "111  https://github.com/jedrake/MEGA                                                           \n",
       "112  https://github.com/lizsuter/Cariaco-NitrogenCycling                                       \n",
       "113  https://github.com/cran/MaxentVariableSelection                                           \n",
       "114  https://github.com/LauraANunes/NunesPearson2016                                           \n",
       "115  https://github.com/cran/rKIN                                                              \n",
       "116  https://github.com/kumarchinnakali/digital-foundry-demand-forcasting                      \n",
       "118  https://github.com/cjbattey/nicheTrackR                                                   \n",
       "119  https://github.com/SEEG-Oxford/ebola_zoonotic                                             \n",
       "120  https://github.com/psesinkclee/biomod2ez                                                  \n",
       "122  https://github.com/KarasiewiczStephane/WitOMI                                             \n",
       "123  https://github.com/hannahlowens/BridgeTree                                                \n",
       "124  https://github.com/Model-R/modelr_pkg                                                     \n",
       "126  https://github.com/biology-examples/nicheModeling                                         \n",
       "127  https://github.com/InesDiogo/NicheRanges                                                  \n",
       "128  https://github.com/mlysy/nicheROVER                                                       \n",
       "129  https://github.com/marissalee/E8-NichePlots                                               \n",
       "130  https://github.com/anwala/NicheSimulation                                                 \n",
       "131  https://github.com/ccmothes/NicheModel                                                    \n",
       "132  https://github.com/Flbl/NicheModelling_FBaletaud                                          \n",
       "133  https://github.com/BlasBenito/NicheModellingGBIF                                          \n",
       "134  https://github.com/JonasGeschke/NicheOverlapR                                             \n",
       "135  https://github.com/aortiz24/Allium_nicheModeling                                          \n",
       "136  https://github.com/rdinnager/nichefillr                                                   \n",
       "137  https://github.com/marlonecobos/nichevol                                                  \n",
       "138  https://github.com/igordot/nichexplorer                                                   \n",
       "139  https://github.com/zeynoon/nichetracking                                                  \n",
       "140  https://github.com/Kgiabou/Michela                                                        \n",
       "141  https://github.com/shinysolutions/galliformes                                             \n",
       "142  https://github.com/nmatzke/phyloENM                                                       \n",
       "143  https://github.com/mattecologist/insect_niche                                             \n",
       "144  https://github.com/romunov/niche_hypotheses                                               \n",
       "145  https://github.com/BrianHayden/Niche_scripts                                              \n",
       "146  https://github.com/mshcheg/DecayNiche                                                     \n",
       "147  https://github.com/erstearns/eco_niche                                                    \n",
       "148  https://github.com/kevinlove/Niche-Modeling-Workshop                                      \n",
       "149  https://github.com/loarie/niche_models                                                    \n",
       "150  https://github.com/Cleonia/GerminationNiche                                               \n",
       "151  https://github.com/jonathanrolland/niche_evolution                                        \n",
       "152  https://github.com/basille/wost-clim-niche                                                \n",
       "153  https://github.com/EmmalineSheahan/Fire_ant_niche_modeling                                \n",
       "154  https://github.com/lucianolasala/Wild-boars                                               \n",
       "155  https://github.com/tufui57/Chionochloa-niche-evolution                                    \n",
       "156  https://github.com/ghuertaramos/ENMOD                                                     \n",
       "157  https://github.com/megansadams/Isotopic-niche-diversity                                   \n",
       "158  https://github.com/cran/EnvNicheR                                                         \n",
       "159  https://github.com/DominiqueGravel/ms_niche_allometric                                    \n",
       "160  https://github.com/rgerlame/niche-evolution-project                                       \n",
       "161  https://github.com/patricia-alcantara-p/mobula_japanica-niche                             \n",
       "162  https://github.com/lindsiebug/Ecological_Niche_Modeling                                   \n",
       "163  https://github.com/rjBadger/niche.modelR                                                  \n",
       "164  https://github.com/bdulken/SVZ_Aging_Niche                                                \n",
       "165  https://github.com/tufui57/Acaena-niche-evolution                                         \n",
       "166  https://github.com/tem11010/Niche-Breadth-Range-Size                                      \n",
       "168  https://github.com/piyalkarum/sp_dist-niche_modeling                                      \n",
       "169  https://github.com/zorrodong/germcell                                                     \n",
       "170  https://github.com/everythinginitsrightplace/Social-Democrats-and-Niche-Parties           \n",
       "171  https://github.com/shobhana1306/Ecological-niche-modelling                                \n",
       "172  https://github.com/jguedes81/asian_niche-1                                                \n",
       "173  https://github.com/cwaldock1/RLS-ThermalNiche                                             \n",
       "174  https://github.com/uzma-researcher/Himalayan-Carex-Climatic-niche-Evolution               \n",
       "175  https://github.com/cwaldock1/RLS-thermal-niche-V2                                         \n",
       "176  https://github.com/TryAgainTryAgain/ENFA                                                  \n",
       "177  https://github.com/marlonecobos/ENM_heterogeneity                                         \n",
       "178  https://github.com/vpellissier/packing                                                    \n",
       "179  https://github.com/roliveros-ramos/kali                                                   \n",
       "180  https://github.com/zipkinlab/Rossman_etal_2016_EcolAndEvol                                \n",
       "181  https://github.com/jacobccooper/trochilidae                                               \n",
       "182  https://github.com/phytomosaic/vuln                                                       \n",
       "183  https://github.com/naturalis/trait-geo-diverse-ungulates                                  \n",
       "184  https://github.com/rebekkaallgayer/Hybrid-niche-modelling-Pacific-Bluefin-tuna--Allgayer  \n",
       "185  https://github.com/alieyres/Chapter2                                                      \n",
       "186  https://github.com/SchlossLab/Jenior_Modeling_mSystems_2017                               \n",
       "187  https://github.com/ryanreisinger/giantPetrels                                             \n",
       "188  https://github.com/SaraVarela/Bucanetes                                                   \n",
       "189  https://github.com/pedroeisenlohr/futurescenarios                                         \n",
       "190  https://github.com/kgturner/FranceCG                                                      \n",
       "191  https://github.com/Pakillo/lump.split.pool.ENM                                            "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# This is a manual step\n",
    "df[df.keywords == 'niche'][['name', 'description', 'url']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "names_to_keep = ['ENiRG', 'MunichBFOR', 'multi-modal-ga', 'fuzzy_habitat_modelling', 'niche_modelling',\n",
    "                'ENMGadgets_orig', 'ENM_TheMetaLand', 'spaa', 'Model-R', 'atlasr', 'marburg_zoonotic', \n",
    "                'anhu', 'Mammal-hypervolumes', 'ACERVNicheSpaceFlorida2011-2015', 'RSFSA_R', 'benmR', 'coastal_ENM',\n",
    "                'ENM_manuals', 'humboldt', 'nppen', 'Ecological-replacement', 'phyloclim', 'CENFA', 'PLNT-model',\n",
    "                'ENM', 'envdianthus', 'MaxentModelEvaluations', 'Strassburg_NEE_ENM', 'recline', 'Merow_et_al_2016_GEB_Minxent_Examples',\n",
    "                'Persistence-Times-Code', 'MaxentVariableSelection', 'NunesPearson2016', 'nicheTrackR', 'ebola_zoonotic',\n",
    "                'biomod2ez', 'modelr_pkg', 'nicheModeling', 'nicheROVER', 'NicheModel', 'NicheModelling_FBaletaud',\n",
    "                'NicheModellingGBIF', 'NicheOverlapR', 'phyloENM', '', 'Niche-Modeling-Workshop', 'GerminationNiche',\n",
    "                'wost-clim-niche', 'Wild-boars', 'ENMOD', 'Isotopic-niche-diversity', 'EnvNicheR', 'mobula_japanica-niche',\n",
    "                'Niche-Breadth-Range-Size', 'Ecological-niche-modelling', 'asian_niche-1', 'Himalayan-Carex-Climatic-niche-Evolution',\n",
    "                'RLS-thermal-niche-V2', 'ENFA', 'ENM_heterogeneity', 'kali', 'Rossman_etal_2016_EcolAndEvol', 'trochilidae',\n",
    "                'vuln', 'trait-geo-diverse-ungulates', 'Hybrid-niche-modelling-Pacific-Bluefin-tuna--Allgayer', 'Chapter2', 'Bucanetes',\n",
    "                'futurescenarios', 'FranceCG', 'lump.split.pool.ENM\t']\n",
    "\n",
    "# must be packages or substantial repos about SDM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "476\n",
      "360\n"
     ]
    }
   ],
   "source": [
    "print(len(df))\n",
    "df_niche = df[(df.name.isin(names_to_keep)) & (df.keywords == 'niche')]\n",
    "df = df[df.keywords != 'niche'].append(df_niche)\n",
    "print(len(df))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Marking repo v pkg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.assign(type_repo = \"package\")\n",
    "\n",
    "df.loc[(~df.description.str.contains('package')) & (df.source == \"Github\"), 'type_repo'] = 'code_base'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>url</th>\n",
       "      <th>keywords</th>\n",
       "      <th>prog_lang</th>\n",
       "      <th>source</th>\n",
       "      <th>stars</th>\n",
       "      <th>duplicate</th>\n",
       "      <th>type_repo</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>biomod2</td>\n",
       "      <td>Ensemble Platform for Species Distribution Modeling</td>\n",
       "      <td>https://cran.r-project.org/web/packages/biomod2/index.html</td>\n",
       "      <td>species distribution</td>\n",
       "      <td>R</td>\n",
       "      <td>CRAN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>package</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>dismo</td>\n",
       "      <td>Species Distribution Modeling</td>\n",
       "      <td>https://cran.r-project.org/web/packages/dismo/index.html</td>\n",
       "      <td>species distribution</td>\n",
       "      <td>R</td>\n",
       "      <td>CRAN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>package</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>fuzzySim</td>\n",
       "      <td>Fuzzy Similarity in Species Distributions</td>\n",
       "      <td>https://cran.r-project.org/web/packages/fuzzySim/index.html</td>\n",
       "      <td>species distribution</td>\n",
       "      <td>R</td>\n",
       "      <td>CRAN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>package</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>gambin</td>\n",
       "      <td>Fit the Gambin Model to Species Abundance Distributions</td>\n",
       "      <td>https://cran.r-project.org/web/packages/gambin/index.html</td>\n",
       "      <td>species distribution</td>\n",
       "      <td>R</td>\n",
       "      <td>CRAN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>package</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hSDM</td>\n",
       "      <td>Hierarchical Bayesian Species Distribution Models</td>\n",
       "      <td>https://cran.r-project.org/web/packages/hSDM/index.html</td>\n",
       "      <td>species distribution</td>\n",
       "      <td>R</td>\n",
       "      <td>CRAN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>package</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       name                                              description  \\\n",
       "0  biomod2   Ensemble Platform for Species Distribution Modeling       \n",
       "1  dismo     Species Distribution Modeling                             \n",
       "2  fuzzySim  Fuzzy Similarity in Species Distributions                 \n",
       "3  gambin    Fit the Gambin Model to Species Abundance Distributions   \n",
       "4  hSDM      Hierarchical Bayesian Species Distribution Models         \n",
       "\n",
       "                                                           url  \\\n",
       "0  https://cran.r-project.org/web/packages/biomod2/index.html    \n",
       "1  https://cran.r-project.org/web/packages/dismo/index.html      \n",
       "2  https://cran.r-project.org/web/packages/fuzzySim/index.html   \n",
       "3  https://cran.r-project.org/web/packages/gambin/index.html     \n",
       "4  https://cran.r-project.org/web/packages/hSDM/index.html       \n",
       "\n",
       "               keywords prog_lang source  stars duplicate type_repo  \n",
       "0  species distribution  R         CRAN  NaN     No        package   \n",
       "1  species distribution  R         CRAN  NaN     No        package   \n",
       "2  species distribution  R         CRAN  NaN     No        package   \n",
       "3  species distribution  R         CRAN  NaN     No        package   \n",
       "4  species distribution  R         CRAN  NaN     No        package   "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sanity checks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>url</th>\n",
       "      <th>keywords</th>\n",
       "      <th>prog_lang</th>\n",
       "      <th>source</th>\n",
       "      <th>stars</th>\n",
       "      <th>duplicate</th>\n",
       "      <th>type_repo</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>iSDM</td>\n",
       "      <td>Invasive Species Distribution Modelling</td>\n",
       "      <td>https://cran.r-project.org/web/packages/iSDM/index.html</td>\n",
       "      <td>species distribution</td>\n",
       "      <td>R</td>\n",
       "      <td>CRAN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>package</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>iNEXT</td>\n",
       "      <td>Interpolation and Extrapolation for Species Diversity</td>\n",
       "      <td>https://cran.r-project.org/web/packages/iNEXT/index.html</td>\n",
       "      <td>species diversity extrapolation</td>\n",
       "      <td>R</td>\n",
       "      <td>CRAN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>package</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303</th>\n",
       "      <td>iSDM</td>\n",
       "      <td>iSDM is an open-source R package that implements a few functions useful for modeling the spatial distribution of inva…</td>\n",
       "      <td>https://github.com/TarekHattab/iSDM</td>\n",
       "      <td>SDM; species distribution</td>\n",
       "      <td>R</td>\n",
       "      <td>Github</td>\n",
       "      <td>7.0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>package</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553</th>\n",
       "      <td>iNEXT</td>\n",
       "      <td>❗️ This is a read-only mirror of the CRAN R package repository. iNEXT — Interpolation and Extrapolation for Species D…</td>\n",
       "      <td>https://github.com/cran/iNEXT</td>\n",
       "      <td>species diversity extrapolation</td>\n",
       "      <td>R</td>\n",
       "      <td>Github</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>package</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      name  \\\n",
       "5    iSDM    \n",
       "39   iNEXT   \n",
       "303  iSDM    \n",
       "553  iNEXT   \n",
       "\n",
       "                                                                                                                description  \\\n",
       "5    Invasive Species Distribution Modelling                                                                                  \n",
       "39   Interpolation and Extrapolation for Species Diversity                                                                    \n",
       "303  iSDM is an open-source R package that implements a few functions useful for modeling the spatial distribution of inva…   \n",
       "553  ❗️ This is a read-only mirror of the CRAN R package repository. iNEXT — Interpolation and Extrapolation for Species D…   \n",
       "\n",
       "                                                          url  \\\n",
       "5    https://cran.r-project.org/web/packages/iSDM/index.html    \n",
       "39   https://cran.r-project.org/web/packages/iNEXT/index.html   \n",
       "303  https://github.com/TarekHattab/iSDM                        \n",
       "553  https://github.com/cran/iNEXT                              \n",
       "\n",
       "                            keywords prog_lang  source  stars duplicate  \\\n",
       "5    species distribution             R         CRAN   NaN     No         \n",
       "39   species diversity extrapolation  R         CRAN   NaN     No         \n",
       "303  SDM; species distribution        R         Github  7.0    Yes        \n",
       "553  species diversity extrapolation  R         Github NaN     Yes        \n",
       "\n",
       "    type_repo  \n",
       "5    package   \n",
       "39   package   \n",
       "303  package   \n",
       "553  package   "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df.name.str.match('^i')] # iNext exists"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>url</th>\n",
       "      <th>keywords</th>\n",
       "      <th>prog_lang</th>\n",
       "      <th>source</th>\n",
       "      <th>stars</th>\n",
       "      <th>duplicate</th>\n",
       "      <th>type_repo</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>iNEXT</td>\n",
       "      <td>Interpolation and Extrapolation for Species Diversity</td>\n",
       "      <td>https://cran.r-project.org/web/packages/iNEXT/index.html</td>\n",
       "      <td>species diversity extrapolation</td>\n",
       "      <td>R</td>\n",
       "      <td>CRAN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>package</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553</th>\n",
       "      <td>iNEXT</td>\n",
       "      <td>❗️ This is a read-only mirror of the CRAN R package repository. iNEXT — Interpolation and Extrapolation for Species D…</td>\n",
       "      <td>https://github.com/cran/iNEXT</td>\n",
       "      <td>species diversity extrapolation</td>\n",
       "      <td>R</td>\n",
       "      <td>Github</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>package</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      name  \\\n",
       "39   iNEXT   \n",
       "553  iNEXT   \n",
       "\n",
       "                                                                                                                description  \\\n",
       "39   Interpolation and Extrapolation for Species Diversity                                                                    \n",
       "553  ❗️ This is a read-only mirror of the CRAN R package repository. iNEXT — Interpolation and Extrapolation for Species D…   \n",
       "\n",
       "                                                          url  \\\n",
       "39   https://cran.r-project.org/web/packages/iNEXT/index.html   \n",
       "553  https://github.com/cran/iNEXT                              \n",
       "\n",
       "                            keywords prog_lang  source  stars duplicate  \\\n",
       "39   species diversity extrapolation  R         CRAN   NaN     No         \n",
       "553  species diversity extrapolation  R         Github NaN     Yes        \n",
       "\n",
       "    type_repo  \n",
       "39   package   \n",
       "553  package   "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df.keywords.str.contains('species diversity extrapolation')]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Summary stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>source</th>\n",
       "      <th>CRAN</th>\n",
       "      <th>Github</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>keywords</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SDM</th>\n",
       "      <td>0</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SDM; species distribution</th>\n",
       "      <td>0</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SDM; species distribution; species range</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>niche</th>\n",
       "      <td>6</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>niche; SDM</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>niche; SDM; species distribution</th>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>niche; species distribution</th>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>niche; species range</th>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>species distribution</th>\n",
       "      <td>25</td>\n",
       "      <td>118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>species distribution; niche</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>species distribution; species occupancy</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>species distribution; species range</th>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>species diversity extrapolation</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>species occupancy</th>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>species occupancy; species range</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>species range</th>\n",
       "      <td>3</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "source                                    CRAN  Github\n",
       "keywords                                              \n",
       "SDM                                       0     42    \n",
       "SDM; species distribution                 0     42    \n",
       "SDM; species distribution; species range  0     1     \n",
       "niche                                     6     68    \n",
       "niche; SDM                                0     1     \n",
       "niche; SDM; species distribution          0     3     \n",
       "niche; species distribution               0     4     \n",
       "niche; species range                      0     2     \n",
       "species distribution                      25    118   \n",
       "species distribution; niche               1     0     \n",
       "species distribution; species occupancy   0     1     \n",
       "species distribution; species range       0     8     \n",
       "species diversity extrapolation           1     1     \n",
       "species occupancy                         2     6     \n",
       "species occupancy; species range          0     1     \n",
       "species range                             3     24    "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.crosstab(df.keywords, df.source)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>source</th>\n",
       "      <th>CRAN</th>\n",
       "      <th>Github</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>type_repo</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>code_base</th>\n",
       "      <td>0</td>\n",
       "      <td>282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>package</th>\n",
       "      <td>38</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "source     CRAN  Github\n",
       "type_repo              \n",
       "code_base  0     282   \n",
       "package    38    40    "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.crosstab(df.type_repo, df.source)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "if 'index' in list(df.columns):\n",
    "    del df['index']\n",
    "\n",
    "df.to_csv(f'data-out/clean/pkgs-comb-{today}.csv', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
